<!DOCTYPE html> <html><head>
		<title>Bioinfo-Matrix-Operations - NOTE</title>
		<base href="../">
		<meta id="root-path" root-path="../">
		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">
		<!--<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes, minimum-scale=1.0, maximum-scale=5.0">-->
		<meta charset="UTF-8">
		<meta name="description" content="Bioinfo-Matrix-Operations - NOTE">
		<meta property="og:title" content="Bioinfo-Matrix-Operations">
		<meta property="og:description" content="Bioinfo-Matrix-Operations - NOTE">
		<meta property="og:type" content="website">
		<meta property="og:url" content="https://www.allenygy.vip/bioinfo/bioinfo-matrix-operations.html">
		<meta property="og:image" content="https://cdn.jsdelivr.net/gh/AllenYGY/ImageSpace@main/uPic/NCz6jn.png">
		<meta property="og:site_name" content="NOTE">
		<meta http-equiv="Cache-Control" content="no-cache, no-store, must-revalidate">
		<meta http-equiv="Pragma" content="no-cache">
		<meta http-equiv="Expires" content="0">
		<meta name="ctime" content="1734159194080">
		<meta name="author" content="AllenYGY"><link rel="alternate" type="application/rss+xml" title="RSS Feed" href="https://www.allenygy.vip/lib/rss.xml"><script async="" id="webpage-script" src="lib/scripts/webpage.js?v=1.8.03" onload="this.onload=null;this.setAttribute(&quot;loaded&quot;, &quot;true&quot;);"></script><script id="minisearch-script" src="https://cdn.jsdelivr.net/npm/minisearch@6.3.0/dist/umd/index.min.js?v=1.8.03" onload="this.onload=null;this.setAttribute(&quot;loaded&quot;, &quot;true&quot;);"></script><script defer="" id="easy-notice-script" src="lib/scripts/easy-notice.js?v=1.8.03" onload="this.onload=null;this.setAttribute(&quot;loaded&quot;, &quot;true&quot;);"></script><link rel="icon" href="lib/media/favicon.png"><style>body{--line-width:40em;--line-width-adaptive:40em;--file-line-width:40em;--sidebar-width:min(20em, 80vw);--collapse-arrow-size:11px;--tree-horizontal-spacing:0.6em;--tree-vertical-spacing:0.6em;--sidebar-margin:12px}.sidebar{height:100%;min-width:calc(var(--sidebar-width) + var(--divider-width-hover));max-width:calc(var(--sidebar-width) + var(--divider-width-hover));font-size:14px;z-index:10;position:relative;overflow:hidden;transition:min-width ease-in-out,max-width ease-in-out;transition-duration:.2s;contain:size}.sidebar-left{left:0}.sidebar-right{right:0}.sidebar.is-collapsed{min-width:0;max-width:0}body.floating-sidebars .sidebar{position:absolute}.sidebar-content{height:100%;min-width:calc(var(--sidebar-width) - var(--divider-width-hover));top:0;padding:var(--sidebar-margin);padding-top:4em;line-height:var(--line-height-tight);background-color:var(--background-secondary);transition:background-color,border-right,border-left,box-shadow;transition-duration:var(--color-fade-speed);transition-timing-function:ease-in-out;position:absolute;display:flex;flex-direction:column}.sidebar:not(.is-collapsed) .sidebar-content{min-width:calc(max(100%,var(--sidebar-width)) - 3px);max-width:calc(max(100%,var(--sidebar-width)) - 3px)}.sidebar-left .sidebar-content{left:0;border-top-right-radius:var(--radius-l);border-bottom-right-radius:var(--radius-l)}.sidebar-right .sidebar-content{right:0;border-top-left-radius:var(--radius-l);border-bottom-left-radius:var(--radius-l)}.sidebar:has(.sidebar-content:empty):has(.topbar-content:empty){display:none}.sidebar-topbar{height:2em;width:var(--sidebar-width);top:var(--sidebar-margin);padding-inline:var(--sidebar-margin);z-index:1;position:fixed;display:flex;align-items:center;transition:width ease-in-out;transition-duration:inherit}.sidebar.is-collapsed .sidebar-topbar{width:calc(2.3em + var(--sidebar-margin) * 2)}.sidebar .sidebar-topbar.is-collapsed{width:0}.sidebar-left .sidebar-topbar{left:0}.sidebar-right .sidebar-topbar{right:0}.topbar-content{overflow:hidden;overflow:clip;width:100%;height:100%;display:flex;align-items:center;transition:inherit}.sidebar.is-collapsed .topbar-content{width:0;transition:inherit}.clickable-icon.sidebar-collapse-icon{background-color:transparent;color:var(--icon-color-focused);padding:0!important;margin:0!important;height:100%!important;width:2.3em!important;margin-inline:0.14em!important;position:absolute}.sidebar-left .clickable-icon.sidebar-collapse-icon{transform:rotateY(180deg);right:var(--sidebar-margin)}.sidebar-right .clickable-icon.sidebar-collapse-icon{transform:rotateY(180deg);left:var(--sidebar-margin)}.clickable-icon.sidebar-collapse-icon svg.svg-icon{width:100%;height:100%}.sidebar-section-header{margin:0 0 1em 0;text-transform:uppercase;letter-spacing:.06em;font-weight:600}body{transition:background-color var(--color-fade-speed) ease-in-out}.webpage-container{display:flex;flex-direction:row;height:100%;width:100%;align-items:stretch;justify-content:center}.document-container{opacity:1;flex-basis:100%;max-width:100%;width:100%;height:100%;display:flex;flex-direction:column;align-items:center;transition:opacity .2s ease-in-out;contain:inline-size}.hide{opacity:0;transition:opacity .2s ease-in-out}.document-container>.markdown-preview-view{margin:var(--sidebar-margin);margin-bottom:0;width:100%;width:-webkit-fill-available;width:-moz-available;width:fill-available;background-color:var(--background-primary);transition:background-color var(--color-fade-speed) ease-in-out;border-top-right-radius:var(--window-radius,var(--radius-m));border-top-left-radius:var(--window-radius,var(--radius-m));overflow-x:hidden!important;overflow-y:auto!important;display:flex!important;flex-direction:column!important;align-items:center!important;contain:inline-size}.document-container>.markdown-preview-view>.markdown-preview-sizer{padding-bottom:80vh!important;width:100%!important;max-width:var(--line-width)!important;flex-basis:var(--line-width)!important;transition:background-color var(--color-fade-speed) ease-in-out;contain:inline-size}.markdown-rendered img:not([width]),.view-content img:not([width]){max-width:100%;outline:0}.document-container>.view-content.embed{display:flex;padding:1em;height:100%;width:100%;align-items:center;justify-content:center}.document-container>.view-content.embed>*{max-width:100%;max-height:100%;object-fit:contain}:has(> :is(.math,table)){overflow-x:auto!important}.document-container>.view-content{overflow-x:auto;contain:content;padding:0;margin:0;height:100%}.scroll-highlight{position:absolute;width:100%;height:100%;pointer-events:none;z-index:1000;background-color:hsla(var(--color-accent-hsl),.25);opacity:0;padding:1em;inset:50%;translate:-50% -50%;border-radius:var(--radius-s)}</style><script defer="">async function loadIncludes(){if("file:"!=location.protocol){let e=document.querySelectorAll("include");for(let t=0;t<e.length;t++){let o=e[t],l=o.getAttribute("src");try{const e=await fetch(l);if(!e.ok){console.log("Could not include file: "+l),o?.remove();continue}let t=await e.text(),n=document.createRange().createContextualFragment(t),i=Array.from(n.children);for(let e of i)e.classList.add("hide"),e.style.transition="opacity 0.5s ease-in-out",setTimeout((()=>{e.classList.remove("hide")}),10);o.before(n),o.remove(),console.log("Included file: "+l)}catch(e){o?.remove(),console.log("Could not include file: "+l,e);continue}}}else{if(document.querySelectorAll("include").length>0){var e=document.createElement("div");e.id="error",e.textContent="Web server exports must be hosted on an http / web server to be viewed correctly.",e.style.position="fixed",e.style.top="50%",e.style.left="50%",e.style.transform="translate(-50%, -50%)",e.style.fontSize="1.5em",e.style.fontWeight="bold",e.style.textAlign="center",document.body.appendChild(e),document.querySelector(".document-container")?.classList.remove("hide")}}}document.addEventListener("DOMContentLoaded",(()=>{loadIncludes()}));let isFileProtocol="file:"==location.protocol;function waitLoadScripts(e,t){let o=e.map((e=>document.getElementById(e+"-script"))),l=0;!function e(){let n=o[l];l++,n&&"true"!=n.getAttribute("loaded")||l<o.length&&e(),l<o.length?n.addEventListener("load",e):t()}()}</script><script defer="" id="img-zoom-script" src="lib/scripts/img-zoom.js?v=1.8.03" onload="this.onload=null;this.setAttribute(&quot;loaded&quot;, &quot;true&quot;);"></script><link rel="stylesheet" href="lib/styles/obsidian.css?v=1.8.03"><link rel="preload" href="lib/styles/other-plugins.css?v=1.8.03" as="style" onload="this.onload=null;this.rel='stylesheet'"><noscript><link rel="stylesheet" href="lib/styles/other-plugins.css?v=1.8.03"></noscript><link rel="stylesheet" href="lib/styles/theme.css?v=1.8.03"><link rel="preload" href="lib/styles/global-variable-styles.css?v=1.8.03" as="style" onload="this.onload=null;this.rel='stylesheet'"><noscript><link rel="stylesheet" href="lib/styles/global-variable-styles.css?v=1.8.03"></noscript><link rel="preload" href="lib/styles/supported-plugins.css?v=1.8.03" as="style" onload="this.onload=null;this.rel='stylesheet'"><noscript><link rel="stylesheet" href="lib/styles/supported-plugins.css?v=1.8.03"></noscript><link rel="preload" href="lib/styles/main-styles.css?v=1.8.03" as="style" onload="this.onload=null;this.rel='stylesheet'"><noscript><link rel="stylesheet" href="lib/styles/main-styles.css?v=1.8.03"></noscript><link rel="preload" href="lib/styles/snippets.css?v=1.8.03" as="style" onload="this.onload=null;this.rel='stylesheet'"><noscript><link rel="stylesheet" href="lib/styles/snippets.css?v=1.8.03"></noscript><script>const LANG="en",VERSION="1.8.03",SHOW_COMMENTS=!0,SHOW_PAGE_HEADER=!0,SHOW_PAGE_FOOTER=!0,SHOW_RSS_FEED=!0,SHOW_SCROLL_TO_TOP=!0,SHOW_FILE_NUMBER=!0,EXPORT_PRESET="website",WALINE_SERVER_URL="https://comments.allenygy.vip/",AUTHOR="AllenYGY",VAULT_TITLE="NOTE",FILE_COLORS={app:"#d7fd6d",CNC:"#8190fd",Database:"#d873fc",English:"#85ffaf",IDE:"#fba69d",LIB:"#d7fd6d","LIB/LVGL":"#ffa07a","LIB/U8G2":"#fa8072",MCU:"#ffc0cb",OS:"#d873fc","OS/RTThread":"#00ccff","app/obsidian":"#8190fd","app/sublime":"#00ccff",SAIYANG:"#00ccff","SAIYANG/新厂区系统集成/network":"#d7fd6d","Private/账号和资产":"#ff6600",Private:"#ff6600","MCU/STM32":"#00ccff","MCU/Renesas":"#ff6600","MCU/Nordic":"#00ffcc","MCU/ESP":"#ff0000",Spirit:"#fba69d","MCU/IC":"#8190fd","MCU/STC51":"#85ffaf","MCU/GD32":"#00ccff","3D":"#00ffcc","3D/Solidworks":"#8190fd",BiliBili:"#d873fc","Computer Organization":"#00ccff","Data Analysis":"#00ffcc","Database Management System":"#00ccff",Math:"#ff6600",Tools:"#00ffcc","Math/Discrete Structure":"#d7fd6d","Math/Linear Algebra":"#d7fd6d","Math/Probability Statistics":"#d7fd6d","computer-organization/Draw/D-SR-Latch.png":"#d7fd6d","tools/Git":"#00ccff","tools/Docker":"#00ccff","tools/Conda":"#00ccff","database-management/MySQL":"#00ccff","database-management/Lecture":"#00ccff",Resume:"#00ffcc","Resume/Resume-template.md":"#00ccff","Data Analysis/Hands on Data Analysis":"#d7fd6d","computer-organization/Lecture":"#00ccff","computer-organization/Lab":"#00ccff",Research:"#00ffcc",Contest:"#d7fd6d","Contest/Leetcode":"#ffa07a","contest/Contest-Overview.md":"#d7fd6d",Web:"#00ffcc","Web/CSS":"#00ccff","Contest/Lanqiao":"#00ccff","Web/Vue":"#00ffcc","tools/Shell":"#00ccff","Computer Network":"#00ccff","Operating System":"#8190fd","Fix Income":"#d7fd6d","Deep Learning":"#00ccff",Others:"#00ffcc",Workflow:"#85ffaf","Web/Javascript":"#fba69d","program/Python":"#00ccff","TaskManagement.md":"#85ffaf",Management:"#8190fd",Algorithm:"#ff6600",Record:"#ff6600",Panel:"#ffa07a","Archive.md":"#ffa07a","Class.md":"#ffa07a","Friends.md":"#00ccff","Workflow/Obsidian/Homepage/Homepage.md":"#8190fd","Task Management.md":"#fa8072","Workflow/Obsidian/obsidian-git":"#d7fd6d","Workflow/Obsidian/Templaters":"#8190fd",Projects:"#8190fd","Slides.md":"#d873fc","Home.md":"#8190fd","Workflow/Obsidian/Homepage":"#8190fd","Machine Learning":"#ffa07a","Computer Vision":"#00ccff","Data Analysis/Big Data Analysis":"#d7fd6d",Bioinfo:"#d7fd6d","Compiler Construction":"#00ccff","Theory Of Computation":"#00ccff",Causality:"#85ffaf","Math/Optimization":"#d7fd6d",Share:"#ff6600"},DEFAULT_THEME="dark",USE_FILE_COLOR_PLUGIN=!0,USE_BARTENDER_PLUGIN=!0</script><script src="https://unpkg.com/prismjs@1.29.0/prism.js" defer="defer"></script><script src="https://unpkg.com/lucide@latest"></script><link rel="stylesheet" href="https://unpkg.com/@waline/client@v3/dist/waline.css"><link rel="stylesheet" href="https://unpkg.com/@waline/client@v3/dist/waline-meta.css"><style>body{--font-ui-medium:16px}:root{--waline-font-size:16px;--waline-white:#fff;--waline-light-grey:#999;--waline-dark-grey:#666;--waline-theme-color:#27ae60;--waline-active-color:#2ecc71;--waline-color:#444;--waline-bg-color:var(--background-primary-alt);--waline-bg-color-light:#f8f8f8;--waline-bg-color-hover:#f0f0f0;--waline-border-color:#ddd;--waline-disable-bg-color:#f8f8f8;--waline-disable-color:#bbb;--waline-code-bg-color:#282c34;--waline-bq-color:#f0f0f0;--waline-avatar-size:3.25rem;--waline-m-avatar-size:calc(var(--waline-avatar-size) * 9 / 13);--waline-badge-color:#3498db;--waline-badge-font-size:0.775em;--waline-info-bg-color:#f8f8f8;--waline-info-color:#999;--waline-info-font-size:0.625em;--waline-border:1px solid var(--waline-border-color);--waline-avatar-radius:50%;--waline-box-shadow:none}.waline-theme-dark{--waline-white:#000;--waline-light-grey:#666;--waline-dark-grey:#999;--waline-color:#888;--waline-bg-color:var(--background-primary-alt);--waline-bg-color-light:#272727;--waline-border-color:#333;--waline-disable-bg-color:#444;--waline-disable-color:#272727;--waline-bq-color:#272727;--waline-info-bg-color:#272727;--waline-info-color:#666}.wl-power{display:none}.wl-reaction{margin-bottom:1em;text-align:left;margin-left:.5em}.wl-reaction-title{margin:20px auto}.wl-reaction-list{justify-content:left}.document-container>.markdown-preview-view>.markdown-preview-sizer{padding-bottom:17px!important}.wl-panel{margin:0}#waline{margin-top:50px!important;max-width:800px}.clickable-icon.sidebar-collapse-icon{width:2em!important;background-color:var(--background-secondary)}.graph-view-placeholder{width:22%;margin-left:38px}.tree-container .clickable-icon{margin-left:2px;border-radius:5px;border:none;box-shadow:none;display:flex;align-items:center;justify-content:center;margin-top:-3px}.document-container>.markdown-preview-view>.markdown-preview-sizer{max-width:100vw!important}body.show-inline-title .page-title{text-align:center}input[type=text],textarea{font-size:16px!important}input[type=search]{width:85%;border:1px solid #888;border-radius:5px;background-color:var(--background-primary);color:var(--text-muted)}.tree-container .clickable-icon{background-color:transparent}body{user-select:unset}blockquote{margin:0}#copyright{text-align:center;margin-top:80px!important}#copyright a{color:#ee802f;text-decoration:none;font-size:15px;margin-left:4px}#copyright a img{width:14px}.document-container>.markdown-preview-view>.markdown-preview-sizer pre{overflow:auto}.theme-light .document-container>.markdown-preview-view>.markdown-preview-sizer pre{position:relative;background-color:var(--background-primary-alt);border:1px solid #eeecec;border-radius:5px;padding:10px}.theme-dark .document-container>.markdown-preview-view>.markdown-preview-sizer pre{position:relative;background-color:var(--background-primary-alt);border:1px solid #373838;border-radius:5px;padding:10px}.markdown-rendered pre button.copy-code-button{position:absolute;top:7px;right:7px}.wl-header input{font-size:16px}.wl-reaction-votes{text-align:center}html>body>.webpage-container>.document-container>.markdown-preview-view>.markdown-preview-sizer>div.page-meta{text-align:center;padding:10px!important}.page-ctime,.waline-comment-count,.waline-pageview-count{margin-right:35px}.page-meta{color:#888}.page-meta svg{width:16px;vertical-align:middle}#qrcode{position:relative}#qr-zoom{position:absolute;width:170px;height:200px;z-index:999;display:none;border:1px solid #ccc;background:#fff}#qrcode:hover #qr-zoom{display:inline-block}#qr-zoom img{margin-top:10px}#qr-zoom font{margin-top:4px;display:inline-block}@media (max-width:768px){#qrcode{display:none!important}}#scrollToTopBtn{display:none;position:fixed;width:46px;height:46px;border:1px solid #ddd;border-radius:5px;background-color:var(--background-primary);text-align:center;bottom:75px;right:306px;opacity:.8}#scrollToTopBtn:hover{background-color:var(--background-primary-alt)}#scrollToTopBtn svg{margin-top:11px;color:var(--text-normal)}#scrollToTopBtn.st-collapsed{right:30px}@media (max-width:768px){#scrollToTopBtn{right:12px}#scrollToTopBtn.st-collapsed{right:12px}}.theme-dark #scrollToTopBtn{border:1px solid #999}.markdown-rendered pre button.copy-code-button{background-color:var(--background-primary);padding:3px 6px;border:1px solid #888;border-radius:5px;opacity:.9}.document-container.markdown-reading-view{background-color:var(--background-primary)}table{border:1px solid var(--table-border-color);border-radius:var(--size-2-3);border-spacing:0}td,th{padding:10px;border-right:1px solid var(--table-border-color);border-bottom:1px solid var(--table-border-color)}td:last-child,th:last-child{border-right:0}tr:last-child td,tr:last-child:not(:only-child) th{border-bottom:0}table>:only-child th{border-bottom:0}.document-container>.markdown-preview-view{margin:0;padding:var(--sidebar-margin)}.sidebar-content{background-color:var(--background-primary-alt)}.nav-folder-title[data-count]::after{content:attr(data-count);font-size:calc(100% * .8);margin-right:12px}.tree-container .tree-header{width:100%}#total-files{position:absolute;right:12px;color:var(--nav-item-color);font-size:calc(100% * .8)}.tree-container .collapse-icon{display:flex;align-items:center}.internal-embed.file-embed{cursor:pointer}.sidebar-content{padding-left:0;padding-right:0}.search-input-container{padding-left:10px}.tree-container{padding-left:12px;padding-right:12px}.graph-view-wrapper{padding-left:12px;padding-right:12px}.tree-container .tree-header,body .webpage-container .tree-container .tree-scroll-area{width:calc(100% - 12px)}.tree-container .clickable-icon svg{width:18px}ul.dataview.list-view-ul li{line-height:180%}#webpage-icon{display:none}.markdown-reading-view table{margin-block-start:var(--p-spacing);margin-block-end:var(--p-spacing)}li{line-height:180%}</style><script type="module">import{init}from"https://unpkg.com/@waline/client@v3/dist/waline.js";window.initWaline=e=>{e=e||document;const t=location.pathname.replace(/index\.html$/,"").replace(/\/$/,"")||"/",o=document.querySelector("#waline"),n=document.querySelector(".markdown-preview-sizer");if(!o){const e=document.createElement("div");e.id="waline",SHOW_COMMENTS||(e.style.display="none"),n.appendChild(e)}let i=document.querySelector(".page-meta");if(!i){const o=document.querySelector(".page-title");i=document.createElement("div"),i.className="page-meta",SHOW_PAGE_HEADER||(i.style.display="none");const n=e.querySelector('meta[name="ctime"]')?.getAttribute("content")||"";let c='<span title="创建时间"><i data-lucide="clock"></i> <span class="page-ctime">'+timestampToTimeString(+n)+"</span></span> ";SHOW_COMMENTS&&(c+='<span title="评论量"><i data-lucide="message-circle-more"></i> <span class="waline-comment-count" data-path="'+t+'">0</span></span> '),c+='<span title="阅读量"><i data-lucide="eye"></i> <span class="waline-pageview-count" data-path="'+t+'">1</span></span> ',c+='<span id="qrcode" title="手机查看"><i data-lucide="qr-code"></i><span id="qr-zoom"><img src="https://api.qrserver.com/v1/create-qr-code/?size=150x150&data='+encodeURIComponent(location.href)+'" /><br /><font>手机查看</font></span></span>',i.innerHTML=c,o.after(i),lucide.createIcons()}try{(SHOW_PAGE_HEADER||SHOW_COMMENTS)&&init({el:"#waline",reaction:!0,dark:!1,copyright:!1,pageview:!0,comment:!0,path:t,serverURL:WALINE_SERVER_URL||"",locale:{reaction0:"点赞",reaction1:"踩一下",reaction2:"酷毙了",reaction3:"不屑",reaction4:"尴尬",reaction5:"无聊"}})}catch(e){console.log(e)}if(SHOW_SCROLL_TO_TOP&&!document.getElementById("scrollToTop")){const e=document.createElement("div");e.id="scrollToTop",e.innerHTML='<div id="scrollToTopBtn"><i data-lucide="arrow-up-to-line"></i></div>',n.appendChild(e),lucide.createIcons();var c=document.querySelector(".document-container>.markdown-preview-view"),a=document.getElementById("scrollToTopBtn");c.addEventListener("scroll",(function(){var e=c.scrollTop,t=.25*c.clientHeight;a.style.display=e>t?"block":"none"})),a.addEventListener("click",(function(){var e=document.querySelector(".document-container>.markdown-preview-view");"scrollBehavior"in document.documentElement.style?e.scrollTo({top:0,behavior:"smooth"}):e.scrollTop=0})),onToggleCollapsedChangedToTop();document.querySelector(".sidebar-right .clickable-icon.sidebar-collapse-icon").addEventListener("click",(()=>{setTimeout((()=>{onToggleCollapsedChangedToTop()}),42)}))}document.querySelectorAll(".document-container>.markdown-preview-view>.markdown-preview-sizer pre").forEach((e=>{e.addEventListener("scroll",(()=>{const t=e.querySelector(".copy-code-button"),o=t.getBoundingClientRect();t.style.right="auto",t.style.left=e.scrollLeft+e.offsetWidth-o.width-8+"px"}))})),setTimeout((()=>{if(!SHOW_COMMENTS)return;const e=document.querySelector(".wl-count");new MutationObserver((e=>{e.forEach((e=>{"characterData"!==e.type&&"childList"!==e.type||e.target.firstChild&&(document.querySelector(".waline-comment-count").innerHTML=e.target.firstChild.textContent)}))})).observe(e,{attributes:!1,childList:!0,subtree:!0,characterData:!0})}),100)}</script><script>function onAsyncLoadedDocument(e,t){togglePrismTheme("dark"!==localStorage.getItem("theme")),setTimeout((()=>{loadSpecialImg()}),42),initWaline(t),copyright(),initImgZoom(),attachDownload()}function toggleWalineTheme(e){e||document.body.classList.contains("waline-theme-dark")?document.body.classList.remove("waline-theme-dark"):document.body.classList.add("waline-theme-dark")}function onThemeToggle(e){toggleWalineTheme(e),togglePrismTheme(e)}function onSidebarCollapsed(e,t,o){const n=o.classList.contains("sidebar-left"),i=document.body.classList.contains("floating-sidebars");if(onToggleCollapsedChangedToTop(),i){const t=document.querySelector("#sidebarMask"),o=document.querySelector(".sidebar-left .tree-scroll-area"),i=isMobile();e?(t&&(t.style.display="none"),n&&o&&i&&(o.style.display="none")):(t&&(t.style.display="block"),n&&o&&i&&setTimeout((()=>{o.style.display="block"}),100))}}function onToggleCollapsedChangedToTop(){if("undefined"!=typeof scrollToTopBtn){document.querySelector(".sidebar-right").classList.contains("is-collapsed")?scrollToTopBtn.classList.add("st-collapsed"):scrollToTopBtn.classList.remove("st-collapsed")}}function copyright(){let e="© "+(new Date).getFullYear()+" "+(AUTHOR||document.querySelector('meta[name="author"]')?.getAttribute("content")||VAULT_TITLE)+' <font color="#ee802f">♡</font> 由obsidian强力驱动';SHOW_RSS_FEED&&(e+='&nbsp;<a href="lib/rss.xml"><img src="https://s21.ax1x.com/2024/04/21/pkpNoSf.png" />RSS</a>');const t=document.querySelector(".markdown-preview-sizer"),o=document.createElement("div");o.id="copyright",SHOW_PAGE_FOOTER||(o.style.display="none"),o.innerHTML=e,t.appendChild(o)}function timestampToTimeString(e){e=e||null;let t=new Date(e);return t.getFullYear()+"-"+((t.getMonth()+1<10?"0"+(t.getMonth()+1):t.getMonth()+1)+"-")+((t.getDate()<10?"0"+t.getDate():t.getDate())+" ")}function togglePrismTheme(e){const t=e?"":"-dark",o=document.getElementById("prism-theme"),n="https://unpkg.com/prismjs@1.29.0/themes/prism"+t+".min.css";if(o&&o.href===n)return;const i=document.createElement("link");i.id="prism-theme",i.rel="stylesheet",i.href=n,i.onload=()=>{o&&o.remove()},document.head.appendChild(i)}function onDomReady(e){"loading"!==document.readyState?e():document.addEventListener("DOMContentLoaded",e)}function isMobileMedia(e=768){return window.matchMedia("(max-width: "+e+"px)").matches}function initImgZoom(){ImgZoom(".document-container>.markdown-preview-view>.markdown-preview-sizer > div:not(.page-meta,.mod-footer,#waline,#scrollToTop,#copyright) img")}function onSearch(){document.querySelector(".sidebar-left input[type=search]").addEventListener("input",(e=>{setTimeout((()=>setTreeCount()),42)}))}function setTreeCount(){const e=document.querySelector(".sidebar-left .file-tree").querySelectorAll(".nav-file:not(.filtered-out)").length;if(document.querySelector("#total-files"))document.querySelector("#total-files").innerHTML=e;else{const t=document.querySelector(".tree-container .tree-header"),o=document.createElement("span");o.id="total-files",o.innerHTML=e,t.appendChild(o)}document.querySelectorAll(".sidebar-left .file-tree .nav-folder:not(.filtered-out) .tree-link .nav-folder-title").forEach((e=>{const t=e.parentElement.nextElementSibling.querySelectorAll(".nav-file:not(.filtered-out)").length;e.dataset.count=t}))}function observeTreeAreaResize(){const e=document.querySelector(".webpage-container .tree-container .tree-scroll-area");new ResizeObserver((()=>{document.querySelector("#total-files").style.right=e.offsetWidth-e.clientWidth+12+"px"})).observe(e)}function onSearchIndexing(){document.querySelector(".sidebar-left input[type=search]").placeholder="zh-cn"===LANG?"正在建立全文索引...":"Full-text indexing..."}function onSearchReady(){document.querySelector(".sidebar-left input[type=search]").placeholder="Search..."}function loadSpecialImg(){document.querySelectorAll("img[src*='+']").forEach((e=>{e.setAttribute("src",URLencode(e.getAttribute("src")))}))}function attachDownload(){setTimeout((()=>{document.querySelectorAll(".internal-embed.file-embed").forEach((e=>{e.onclick=()=>{window.open(e.getAttribute("src"))}}))}),42)}function onScrollIntoViewStart(e,t,o){setTimeout((()=>{const e=document.querySelector(".document-container>.markdown-preview-view");e.scrollLeft>0&&e.scrollTo(0,e.scrollTop)}),1500)}togglePrismTheme("dark"!==localStorage.getItem("theme")),onDomReady((()=>{let e=0;const t=(o=50)=>{const n=document.querySelector(".mod-tree-file .tree-link[href='index.html'] .tree-item-title");if(!(e>6e4)){if(n&&"index"===n.innerHTML)return n.innerHTML="首页",void loadSpecialImg();e+=o,setTimeout((()=>{t(o)}),o)}};"zh-cn"===LANG&&t(),initWaline(),toggleWalineTheme("dark"!==localStorage.getItem("theme")),copyright(),document.querySelector(".document-container").classList.remove("hide");const o=setInterval((()=>{const e=document.querySelector(".nav-file.mod-active");e&&e.parentElement&&(o&&clearInterval(o),setTimeout((()=>{if(setTimeout((()=>{const e=document.querySelectorAll(".tree-container .collapse-icon");e.length>0&&void 0===e[0].style.translate&&e.forEach((e=>{e.style.left=0}))}),50),SHOW_FILE_NUMBER&&(setTreeCount(),observeTreeAreaResize(),onSearch()),FILE_COLORS&&USE_FILE_COLOR_PLUGIN&&Object.getOwnPropertyNames(FILE_COLORS).length)for(const e in FILE_COLORS){const t=document.querySelector(".sidebar-left .tree-link[data-path='"+e+"'] .tree-item-title");t&&(t.style.color=FILE_COLORS[e])}attachDownload()}),42))}),50),n=document.createElement("div");n.id="sidebarMask",n.style="position:fixed;top:0;left:0;opacity:0;z-index:9;width:100%;height:100%;display:none;",n.onclick=function(){this.style.display="none"},document.body.appendChild(n),initImgZoom(),document.querySelector(".webpage-container input[type=search]").addEventListener("input",(e=>{searchReady||EasyNotice&&EasyNotice.createNotice("zh-cn"===LANG?"正在建立全文索引，请稍后重试...":"Full-text indexing, please try again later.",1500,"left:12px;top:12px;",!0,!1)}))}))</script><script async="" src="https://www.googletagmanager.com/gtag/js?id=G-EJ4LL9C0ZB"></script><script>function gtag(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","G-EJ4LL9C0ZB")</script></head><body class="publish css-settings-manager theme-dark native-scrollbars show-inline-title show-ribbon outliner-plugin-dnd primary-accent-purple secondary-accent-green"><script defer="">let theme=localStorage.getItem("theme")||DEFAULT_THEME||(window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light");"dark"==theme?(document.body.classList.add("theme-dark"),document.body.classList.remove("theme-light")):(document.body.classList.add("theme-light"),document.body.classList.remove("theme-dark")),window.innerWidth<480?document.body.classList.add("is-phone"):window.innerWidth<768?document.body.classList.add("is-tablet"):window.innerWidth<1024?document.body.classList.add("is-small-screen"):document.body.classList.add("is-large-screen")</script><div class="webpage-container workspace"><div class="sidebar-left sidebar"><div class="sidebar-handle"></div><div class="sidebar-topbar"><div class="topbar-content"><label class="theme-toggle-container" for="theme_toggle"><input class="theme-toggle-input" type="checkbox" id="theme_toggle"><div class="toggle-background"></div></label></div><div class="clickable-icon sidebar-collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="100%" height="100%" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon"><path d="M21 3H3C1.89543 3 1 3.89543 1 5V19C1 20.1046 1.89543 21 3 21H21C22.1046 21 23 20.1046 23 19V5C23 3.89543 22.1046 3 21 3Z"></path><path d="M10 4V20"></path><path d="M4 7H7"></path><path d="M4 10H7"></path><path d="M4 13H7"></path></svg></div></div><div class="sidebar-content"><div class="search-input-container"><input enterkeyhint="search" type="search" spellcheck="false" placeholder="Search..."><div class="search-input-clear-button" aria-label="Clear search"></div></div><include src="lib/html/file-tree.html?v=1.8.03&amp;t=1735216518082"></include></div><script defer="">let ls = document.querySelector(".sidebar-left"); ls.classList.add("is-collapsed"); if (window.innerWidth > 768) ls.classList.remove("is-collapsed"); ls.style.setProperty("--sidebar-width", localStorage.getItem("sidebar-left-width"));</script></div><div class="document-container markdown-reading-view hide"><div class="markdown-preview-view markdown-rendered allow-fold-headings allow-fold-lists is-readable-line-width"><style id="MJX-CHTML-styles">mjx-c.mjx-c1D43B.TEX-I::before { padding: 0.683em 0.888em 0px 0px; content: "H"; }
mjx-c.mjx-c1D44B.TEX-I::before { padding: 0.683em 0.852em 0px 0px; content: "X"; }
mjx-c.mjx-c1D70E.TEX-I::before { padding: 0.431em 0.571em 0.011em 0px; content: "σ"; }
mjx-c.mjx-c75::before { padding: 0.442em 0.556em 0.011em 0px; content: "u"; }
mjx-c.mjx-c1D444.TEX-I::before { padding: 0.704em 0.791em 0.194em 0px; content: "Q"; }
mjx-c.mjx-c1D70C.TEX-I::before { padding: 0.442em 0.517em 0.216em 0px; content: "ρ"; }
mjx-c.mjx-c1D43E.TEX-I::before { padding: 0.683em 0.889em 0px 0px; content: "K"; }
mjx-c.mjx-c1D6FE.TEX-I::before { padding: 0.441em 0.543em 0.216em 0px; content: "γ"; }
mjx-stretchy-v.mjx-c7B mjx-beg mjx-c::before { content: "⎧"; padding: 0.899em 0.889em 0.01em 0px; }
mjx-stretchy-v.mjx-c7B mjx-ext mjx-c::before { content: "⎪"; width: 0.889em; }
mjx-stretchy-v.mjx-c7B mjx-end mjx-c::before { content: "⎩"; padding: 0.01em 0.889em 0.899em 0px; }
mjx-stretchy-v.mjx-c7B mjx-mid mjx-c::before { content: "⎨"; padding: 1.16em 0.889em 0.66em 0px; }
mjx-stretchy-v.mjx-c7B > mjx-mid { margin-top: -0.91em; margin-bottom: -0.91em; }
mjx-stretchy-v.mjx-c7B > mjx-end { margin-top: -0.909em; }
mjx-stretchy-v.mjx-c7B > mjx-ext { height: 50%; border-top-width: 0.879em; border-bottom-width: 0.879em; }
mjx-c.mjx-c2026::before { padding: 0.12em 1.172em 0px 0px; content: "…"; }
mjx-c.mjx-c2032::before { padding: 0.56em 0.275em 0px 0px; content: "′"; }
mjx-c.mjx-c26::before { padding: 0.716em 0.778em 0.022em 0px; content: "&"; }
mjx-msqrt { display: inline-block; text-align: left; }
mjx-root { display: inline-block; white-space: nowrap; }
mjx-surd { display: inline-block; vertical-align: top; }
mjx-sqrt { display: inline-block; padding-top: 0.07em; }
mjx-sqrt > mjx-box { border-top: 0.07em solid; }
mjx-sqrt.mjx-tall > mjx-box { padding-left: 0.3em; margin-left: -0.3em; }
mjx-c.mjx-c3A9::before { padding: 0.704em 0.722em 0px 0px; content: "Ω"; }
mjx-c.mjx-c221A::before { padding: 0.8em 0.853em 0.2em 0px; content: "√"; }
mjx-c.mjx-c2192::before { padding: 0.511em 1em 0.011em 0px; content: "→"; }
mjx-c.mjx-c230A.TEX-S2::before { padding: 1.15em 0.528em 0.649em 0px; content: "⌊"; }
mjx-c.mjx-c53::before { padding: 0.705em 0.556em 0.022em 0px; content: "S"; }
mjx-c.mjx-c7A::before { padding: 0.431em 0.444em 0px 0px; content: "z"; }
mjx-c.mjx-c65::before { padding: 0.448em 0.444em 0.011em 0px; content: "e"; }
mjx-c.mjx-c20::before { padding: 0px 0.25em 0px 0px; content: " "; }
mjx-c.mjx-c6E::before { padding: 0.442em 0.556em 0px 0px; content: "n"; }
mjx-c.mjx-c62::before { padding: 0.694em 0.556em 0.011em 0px; content: "b"; }
mjx-c.mjx-c63::before { padding: 0.448em 0.444em 0.011em 0px; content: "c"; }
mjx-c.mjx-c6B::before { padding: 0.694em 0.528em 0px 0px; content: "k"; }
mjx-c.mjx-c68::before { padding: 0.694em 0.556em 0px 0px; content: "h"; }
mjx-c.mjx-c70::before { padding: 0.442em 0.556em 0.194em 0px; content: "p"; }
mjx-c.mjx-c79::before { padding: 0.431em 0.528em 0.204em 0px; content: "y"; }
mjx-c.mjx-c72::before { padding: 0.442em 0.392em 0px 0px; content: "r"; }
mjx-c.mjx-c64::before { padding: 0.694em 0.556em 0.011em 0px; content: "d"; }
mjx-c.mjx-c230B.TEX-S2::before { padding: 1.15em 0.528em 0.649em 0px; content: "⌋"; }
mjx-c.mjx-c230A.TEX-S1::before { padding: 0.85em 0.472em 0.349em 0px; content: "⌊"; }
mjx-c.mjx-c230B.TEX-S1::before { padding: 0.85em 0.472em 0.349em 0px; content: "⌋"; }
mjx-c.mjx-c74::before { padding: 0.615em 0.389em 0.01em 0px; content: "t"; }
mjx-c.mjx-c1D44A.TEX-I::before { padding: 0.683em 1.048em 0.022em 0px; content: "W"; }
mjx-c.mjx-c2235.TEX-A::before { padding: 0.471em 0.667em 0.082em 0px; content: "∵"; }
mjx-c.mjx-c1D435.TEX-I::before { padding: 0.683em 0.759em 0px 0px; content: "B"; }
mjx-c.mjx-c36::before { padding: 0.666em 0.5em 0.022em 0px; content: "6"; }
mjx-c.mjx-c37::before { padding: 0.676em 0.5em 0.022em 0px; content: "7"; }
mjx-c.mjx-c38::before { padding: 0.666em 0.5em 0.022em 0px; content: "8"; }
mjx-c.mjx-c2308::before { padding: 0.75em 0.444em 0.25em 0px; content: "⌈"; }
mjx-c.mjx-c2309::before { padding: 0.75em 0.444em 0.25em 0px; content: "⌉"; }
mjx-c.mjx-c2234.TEX-A::before { padding: 0.471em 0.667em 0.082em 0px; content: "∴"; }
mjx-c.mjx-c28.TEX-S2::before { padding: 1.15em 0.597em 0.649em 0px; content: "("; }
mjx-c.mjx-c29.TEX-S2::before { padding: 1.15em 0.597em 0.649em 0px; content: ")"; }
mjx-c.mjx-c28.TEX-S3::before { padding: 1.45em 0.736em 0.949em 0px; content: "("; }
mjx-c.mjx-c29.TEX-S3::before { padding: 1.45em 0.736em 0.949em 0px; content: ")"; }
mjx-c.mjx-c394::before { padding: 0.716em 0.833em 0px 0px; content: "Δ"; }
mjx-c.mjx-c221E::before { padding: 0.442em 1em 0.011em 0px; content: "∞"; }
mjx-c.mjx-c2248::before { padding: 0.483em 0.778em 0px 0px; content: "≈"; }
mjx-c.mjx-c6D::before { padding: 0.442em 0.833em 0px 0px; content: "m"; }
mjx-c.mjx-c61::before { padding: 0.448em 0.5em 0.011em 0px; content: "a"; }
mjx-c.mjx-c78::before { padding: 0.431em 0.528em 0px 0px; content: "x"; }
mjx-c.mjx-c1D45E.TEX-I::before { padding: 0.442em 0.46em 0.194em 0px; content: "q"; }
mjx-c.mjx-c1D436.TEX-I::before { padding: 0.705em 0.76em 0.022em 0px; content: "C"; }
mjx-c.mjx-c34::before { padding: 0.677em 0.5em 0px 0px; content: "4"; }
mjx-c.mjx-c3A::before { padding: 0.43em 0.278em 0px 0px; content: ":"; }
mjx-c.mjx-c1D703.TEX-I::before { padding: 0.705em 0.469em 0.01em 0px; content: "θ"; }
mjx-c.mjx-c2200::before { padding: 0.694em 0.556em 0.022em 0px; content: "∀"; }
mjx-c.mjx-c2032::before { padding: 0.56em 0.275em 0px 0px; content: "′"; }
mjx-c.mjx-c1D445.TEX-I::before { padding: 0.683em 0.759em 0.021em 0px; content: "R"; }
mjx-c.mjx-c2211.TEX-S1::before { padding: 0.75em 1.056em 0.25em 0px; content: "∑"; }
mjx-c.mjx-c2033::before { padding: 0.56em 0.55em 0px 0px; content: "′′"; }
mjx-c.mjx-c1D458.TEX-I::before { padding: 0.694em 0.521em 0.011em 0px; content: "k"; }
mjx-mrow { display: inline-block; text-align: left; }
mjx-mtable { display: inline-block; text-align: center; vertical-align: 0.25em; position: relative; box-sizing: border-box; border-spacing: 0px; border-collapse: collapse; }
mjx-mstyle[size="s"] mjx-mtable { vertical-align: 0.354em; }
mjx-labels { position: absolute; left: 0px; top: 0px; }
mjx-table { display: inline-block; vertical-align: -0.5ex; box-sizing: border-box; }
mjx-table > mjx-itable { vertical-align: middle; text-align: left; box-sizing: border-box; }
mjx-labels > mjx-itable { position: absolute; top: 0px; }
mjx-mtable[justify="left"] { text-align: left; }
mjx-mtable[justify="right"] { text-align: right; }
mjx-mtable[justify="left"][side="left"] { padding-right: 0px !important; }
mjx-mtable[justify="left"][side="right"] { padding-left: 0px !important; }
mjx-mtable[justify="right"][side="left"] { padding-right: 0px !important; }
mjx-mtable[justify="right"][side="right"] { padding-left: 0px !important; }
mjx-mtable[align] { vertical-align: baseline; }
mjx-mtable[align="top"] > mjx-table { vertical-align: top; }
mjx-mtable[align="bottom"] > mjx-table { vertical-align: bottom; }
mjx-mtable[side="right"] mjx-labels { min-width: 100%; }
mjx-mtr { display: table-row; text-align: left; }
mjx-mtr[rowalign="top"] > mjx-mtd { vertical-align: top; }
mjx-mtr[rowalign="center"] > mjx-mtd { vertical-align: middle; }
mjx-mtr[rowalign="bottom"] > mjx-mtd { vertical-align: bottom; }
mjx-mtr[rowalign="baseline"] > mjx-mtd { vertical-align: baseline; }
mjx-mtr[rowalign="axis"] > mjx-mtd { vertical-align: 0.25em; }
mjx-mtd { display: table-cell; text-align: center; padding: 0.215em 0.4em; }
mjx-mtd:first-child { padding-left: 0px; }
mjx-mtd:last-child { padding-right: 0px; }
mjx-mtable > * > mjx-itable > :first-child > mjx-mtd { padding-top: 0px; }
mjx-mtable > * > mjx-itable > :last-child > mjx-mtd { padding-bottom: 0px; }
mjx-tstrut { display: inline-block; height: 1em; vertical-align: -0.25em; }
mjx-labels[align="left"] > mjx-mtr > mjx-mtd { text-align: left; }
mjx-labels[align="right"] > mjx-mtr > mjx-mtd { text-align: right; }
mjx-mtd[extra] { padding: 0px; }
mjx-mtd[rowalign="top"] { vertical-align: top; }
mjx-mtd[rowalign="center"] { vertical-align: middle; }
mjx-mtd[rowalign="bottom"] { vertical-align: bottom; }
mjx-mtd[rowalign="baseline"] { vertical-align: baseline; }
mjx-mtd[rowalign="axis"] { vertical-align: 0.25em; }
mjx-mfrac { display: inline-block; text-align: left; }
mjx-frac { display: inline-block; vertical-align: 0.17em; padding: 0px 0.22em; }
mjx-frac[type="d"] { vertical-align: 0.04em; }
mjx-frac[delims] { padding: 0px 0.1em; }
mjx-frac[atop] { padding: 0px 0.12em; }
mjx-frac[atop][delims] { padding: 0px; }
mjx-dtable { display: inline-table; width: 100%; }
mjx-dtable > * { font-size: 2000%; }
mjx-dbox { display: block; font-size: 5%; }
mjx-num { display: block; text-align: center; }
mjx-den { display: block; text-align: center; }
mjx-mfrac[bevelled] > mjx-num { display: inline-block; }
mjx-mfrac[bevelled] > mjx-den { display: inline-block; }
mjx-den[align="right"], mjx-num[align="right"] { text-align: right; }
mjx-den[align="left"], mjx-num[align="left"] { text-align: left; }
mjx-nstrut { display: inline-block; height: 0.054em; width: 0px; vertical-align: -0.054em; }
mjx-nstrut[type="d"] { height: 0.217em; vertical-align: -0.217em; }
mjx-dstrut { display: inline-block; height: 0.505em; width: 0px; }
mjx-dstrut[type="d"] { height: 0.726em; }
mjx-line { display: block; box-sizing: border-box; min-height: 1px; height: 0.06em; border-top: 0.06em solid; margin: 0.06em -0.1em; overflow: hidden; }
mjx-line[type="d"] { margin: 0.18em -0.1em; }
mjx-mspace { display: inline-block; text-align: left; }
mjx-munderover { display: inline-block; text-align: left; }
mjx-munderover:not([limits="false"]) { padding-top: 0.1em; }
mjx-munderover:not([limits="false"]) > * { display: block; }
mjx-msubsup { display: inline-block; text-align: left; }
mjx-script { display: inline-block; padding-right: 0.05em; padding-left: 0.033em; }
mjx-script > mjx-spacer { display: block; }
mjx-c.mjx-c7B.TEX-S3::before { padding: 1.45em 0.75em 0.949em 0px; content: "{"; }
mjx-c.mjx-c24::before { padding: 0.75em 0.5em 0.056em 0px; content: "$"; }
mjx-c.mjx-c69::before { padding: 0.669em 0.278em 0px 0px; content: "i"; }
mjx-c.mjx-c66::before { padding: 0.705em 0.372em 0px 0px; content: "f"; }
mjx-c.mjx-c2217::before { padding: 0.465em 0.5em 0px 0px; content: "∗"; }
mjx-c.mjx-c28.TEX-S1::before { padding: 0.85em 0.458em 0.349em 0px; content: "("; }
mjx-c.mjx-c29.TEX-S1::before { padding: 0.85em 0.458em 0.349em 0px; content: ")"; }
mjx-c.mjx-c5B.TEX-S1::before { padding: 0.85em 0.417em 0.349em 0px; content: "["; }
mjx-c.mjx-c5D.TEX-S1::before { padding: 0.85em 0.417em 0.349em 0px; content: "]"; }
mjx-c.mjx-c6C::before { padding: 0.694em 0.278em 0px 0px; content: "l"; }
mjx-c.mjx-c6F::before { padding: 0.448em 0.5em 0.01em 0px; content: "o"; }
mjx-c.mjx-c67::before { padding: 0.453em 0.5em 0.206em 0px; content: "g"; }
mjx-c.mjx-c2061::before { padding: 0px; content: ""; }
mjx-c.mjx-c22C5::before { padding: 0.31em 0.278em 0px 0px; content: "⋅"; }
mjx-c.mjx-c230A::before { padding: 0.75em 0.444em 0.25em 0px; content: "⌊"; }
mjx-c.mjx-c230B::before { padding: 0.75em 0.444em 0.25em 0px; content: "⌋"; }
mjx-msup { display: inline-block; text-align: left; }
mjx-texatom { display: inline-block; text-align: left; }
mjx-munder { display: inline-block; text-align: left; }
mjx-over { text-align: left; }
mjx-munder:not([limits="false"]) { display: inline-table; }
mjx-munder > mjx-row { text-align: left; }
mjx-under { padding-bottom: 0.1em; }
mjx-c.mjx-c1D43A.TEX-I::before { padding: 0.705em 0.786em 0.022em 0px; content: "G"; }
mjx-c.mjx-c1D449.TEX-I::before { padding: 0.683em 0.769em 0.022em 0px; content: "V"; }
mjx-c.mjx-c1D438.TEX-I::before { padding: 0.68em 0.764em 0px 0px; content: "E"; }
mjx-c.mjx-c2208::before { padding: 0.54em 0.667em 0.04em 0px; content: "∈"; }
mjx-c.mjx-c5B::before { padding: 0.75em 0.278em 0.25em 0px; content: "["; }
mjx-c.mjx-c5D::before { padding: 0.75em 0.278em 0.25em 0px; content: "]"; }
mjx-c.mjx-c7B::before { padding: 0.75em 0.5em 0.25em 0px; content: "{"; }
mjx-c.mjx-c33::before { padding: 0.665em 0.5em 0.022em 0px; content: "3"; }
mjx-c.mjx-c39::before { padding: 0.666em 0.5em 0.022em 0px; content: "9"; }
mjx-c.mjx-c7D::before { padding: 0.75em 0.5em 0.25em 0px; content: "}"; }
mjx-c.mjx-c32::before { padding: 0.666em 0.5em 0px 0px; content: "2"; }
mjx-c.mjx-c1D463.TEX-I::before { padding: 0.443em 0.485em 0.011em 0px; content: "v"; }
mjx-c.mjx-c7C::before { padding: 0.75em 0.278em 0.249em 0px; content: "|"; }
mjx-c.mjx-c1D434.TEX-I::before { padding: 0.716em 0.75em 0px 0px; content: "A"; }
mjx-c.mjx-cD7::before { padding: 0.491em 0.778em 0px 0px; content: "×"; }
mjx-c.mjx-c1D450.TEX-I::before { padding: 0.442em 0.433em 0.011em 0px; content: "c"; }
mjx-c.mjx-c1D453.TEX-I::before { padding: 0.705em 0.55em 0.205em 0px; content: "f"; }
mjx-c.mjx-c1D439.TEX-I::before { padding: 0.68em 0.749em 0px 0px; content: "F"; }
mjx-c.mjx-c2260::before { padding: 0.716em 0.778em 0.215em 0px; content: "≠"; }
mjx-c.mjx-c1D448.TEX-I::before { padding: 0.683em 0.767em 0.022em 0px; content: "U"; }
mjx-c.mjx-c1D43F.TEX-I::before { padding: 0.683em 0.681em 0px 0px; content: "L"; }
mjx-c.mjx-c2211.TEX-S2::before { padding: 0.95em 1.444em 0.45em 0px; content: "∑"; }
mjx-c.mjx-c1D447.TEX-I::before { padding: 0.677em 0.704em 0px 0px; content: "T"; }
mjx-c.mjx-c1D437.TEX-I::before { padding: 0.683em 0.828em 0px 0px; content: "D"; }
mjx-c.mjx-c1D446.TEX-I::before { padding: 0.705em 0.645em 0.022em 0px; content: "S"; }
mjx-c.mjx-c1D442.TEX-I::before { padding: 0.704em 0.763em 0.022em 0px; content: "O"; }
mjx-c.mjx-c21D2::before { padding: 0.525em 1em 0.024em 0px; content: "⇒"; }
mjx-msub { display: inline-block; text-align: left; }
mjx-mtext { display: inline-block; text-align: left; }
mjx-c.mjx-c1D466.TEX-I::before { padding: 0.442em 0.49em 0.205em 0px; content: "y"; }
mjx-c.mjx-c1D467.TEX-I::before { padding: 0.442em 0.465em 0.011em 0px; content: "z"; }
mjx-c.mjx-c2C::before { padding: 0.121em 0.278em 0.194em 0px; content: ","; }
mjx-c.mjx-c1D45A.TEX-I::before { padding: 0.442em 0.878em 0.011em 0px; content: "m"; }
mjx-c.mjx-c1D45D.TEX-I::before { padding: 0.442em 0.503em 0.194em 0px; content: "p"; }
mjx-c.mjx-c35::before { padding: 0.666em 0.5em 0.022em 0px; content: "5"; }
mjx-c.mjx-c30::before { padding: 0.666em 0.5em 0.022em 0px; content: "0"; }
mjx-c.mjx-c1D440.TEX-I::before { padding: 0.683em 1.051em 0px 0px; content: "M"; }
mjx-c.mjx-c1D456.TEX-I::before { padding: 0.661em 0.345em 0.011em 0px; content: "i"; }
mjx-c.mjx-c1D457.TEX-I::before { padding: 0.661em 0.412em 0.204em 0px; content: "j"; }
mjx-c.mjx-cA0::before { padding: 0px 0.25em 0px 0px; content: " "; }
mjx-c.mjx-c1D441.TEX-I::before { padding: 0.683em 0.888em 0px 0px; content: "N"; }
mjx-c.mjx-c1D443.TEX-I::before { padding: 0.683em 0.751em 0px 0px; content: "P"; }
mjx-c.mjx-c1D44E.TEX-I::before { padding: 0.441em 0.529em 0.01em 0px; content: "a"; }
mjx-c.mjx-c1D460.TEX-I::before { padding: 0.442em 0.469em 0.01em 0px; content: "s"; }
mjx-c.mjx-c2E::before { padding: 0.12em 0.278em 0px 0px; content: "."; }
mjx-c.mjx-c1D454.TEX-I::before { padding: 0.442em 0.477em 0.205em 0px; content: "g"; }
mjx-c.mjx-c1D461.TEX-I::before { padding: 0.626em 0.361em 0.011em 0px; content: "t"; }
mjx-c.mjx-c210E.TEX-I::before { padding: 0.694em 0.576em 0.011em 0px; content: "h"; }
mjx-c.mjx-c3D::before { padding: 0.583em 0.778em 0.082em 0px; content: "="; }
mjx-math { display: inline-block; text-align: left; line-height: 0; text-indent: 0px; font-style: normal; font-weight: normal; font-size: 100%; letter-spacing: normal; border-collapse: collapse; overflow-wrap: normal; word-spacing: normal; white-space: nowrap; direction: ltr; padding: 1px 0px; }
mjx-container[jax="CHTML"][display="true"] { display: block; text-align: center; margin: 1em 0px; }
mjx-container[jax="CHTML"][display="true"][width="full"] { display: flex; }
mjx-container[jax="CHTML"][display="true"] mjx-math { padding: 0px; }
mjx-container[jax="CHTML"][justify="left"] { text-align: left; }
mjx-container[jax="CHTML"][justify="right"] { text-align: right; }
mjx-mo { display: inline-block; text-align: left; }
mjx-stretchy-h { display: inline-table; width: 100%; }
mjx-stretchy-h > * { display: table-cell; width: 0px; }
mjx-stretchy-h > * > mjx-c { display: inline-block; transform: scaleX(1); }
mjx-stretchy-h > * > mjx-c::before { display: inline-block; width: initial; }
mjx-stretchy-h > mjx-ext { overflow: clip visible; width: 100%; }
mjx-stretchy-h > mjx-ext > mjx-c::before { transform: scaleX(500); }
mjx-stretchy-h > mjx-ext > mjx-c { width: 0px; }
mjx-stretchy-h > mjx-beg > mjx-c { margin-right: -0.1em; }
mjx-stretchy-h > mjx-end > mjx-c { margin-left: -0.1em; }
mjx-stretchy-v { display: inline-block; }
mjx-stretchy-v > * { display: block; }
mjx-stretchy-v > mjx-beg { height: 0px; }
mjx-stretchy-v > mjx-end > mjx-c { display: block; }
mjx-stretchy-v > * > mjx-c { transform: scaleY(1); transform-origin: left center; overflow: hidden; }
mjx-stretchy-v > mjx-ext { display: block; height: 100%; box-sizing: border-box; border: 0px solid transparent; overflow: visible clip; }
mjx-stretchy-v > mjx-ext > mjx-c::before { width: initial; box-sizing: border-box; }
mjx-stretchy-v > mjx-ext > mjx-c { transform: scaleY(500) translateY(0.075em); overflow: visible; }
mjx-mark { display: inline-block; height: 0px; }
mjx-c { display: inline-block; }
mjx-utext { display: inline-block; padding: 0.75em 0px 0.2em; }
mjx-mi { display: inline-block; text-align: left; }
mjx-mn { display: inline-block; text-align: left; }
mjx-c.mjx-c2265::before { padding: 0.636em 0.778em 0.138em 0px; content: "≥"; }
mjx-c.mjx-c1D459.TEX-I::before { padding: 0.694em 0.298em 0.011em 0px; content: "l"; }
mjx-c.mjx-c1D45C.TEX-I::before { padding: 0.441em 0.485em 0.011em 0px; content: "o"; }
mjx-c.mjx-c1D464.TEX-I::before { padding: 0.443em 0.716em 0.011em 0px; content: "w"; }
mjx-c.mjx-c1D452.TEX-I::before { padding: 0.442em 0.466em 0.011em 0px; content: "e"; }
mjx-c.mjx-c1D45F.TEX-I::before { padding: 0.442em 0.451em 0.011em 0px; content: "r"; }
mjx-c.mjx-c1D44F.TEX-I::before { padding: 0.694em 0.429em 0.011em 0px; content: "b"; }
mjx-c.mjx-c1D462.TEX-I::before { padding: 0.442em 0.572em 0.011em 0px; content: "u"; }
mjx-c.mjx-c1D45B.TEX-I::before { padding: 0.442em 0.6em 0.011em 0px; content: "n"; }
mjx-c.mjx-c1D451.TEX-I::before { padding: 0.694em 0.52em 0.01em 0px; content: "d"; }
mjx-c.mjx-c3E::before { padding: 0.54em 0.778em 0.04em 0px; content: ">"; }
mjx-c.mjx-c28::before { padding: 0.75em 0.389em 0.25em 0px; content: "("; }
mjx-c.mjx-c1D465.TEX-I::before { padding: 0.442em 0.572em 0.011em 0px; content: "x"; }
mjx-c.mjx-c2B::before { padding: 0.583em 0.778em 0.082em 0px; content: "+"; }
mjx-c.mjx-c31::before { padding: 0.666em 0.5em 0px 0px; content: "1"; }
mjx-c.mjx-c29::before { padding: 0.75em 0.389em 0.25em 0px; content: ")"; }
mjx-c.mjx-c3C::before { padding: 0.54em 0.778em 0.04em 0px; content: "<"; }
mjx-c.mjx-c2212::before { padding: 0.583em 0.778em 0.082em 0px; content: "−"; }
mjx-c.mjx-c2264::before { padding: 0.636em 0.778em 0.138em 0px; content: "≤"; }
mjx-container[jax="CHTML"] { line-height: 0; }
mjx-container [space="1"] { margin-left: 0.111em; }
mjx-container [space="2"] { margin-left: 0.167em; }
mjx-container [space="3"] { margin-left: 0.222em; }
mjx-container [space="4"] { margin-left: 0.278em; }
mjx-container [space="5"] { margin-left: 0.333em; }
mjx-container [rspace="1"] { margin-right: 0.111em; }
mjx-container [rspace="2"] { margin-right: 0.167em; }
mjx-container [rspace="3"] { margin-right: 0.222em; }
mjx-container [rspace="4"] { margin-right: 0.278em; }
mjx-container [rspace="5"] { margin-right: 0.333em; }
mjx-container [size="s"] { font-size: 70.7%; }
mjx-container [size="ss"] { font-size: 50%; }
mjx-container [size="Tn"] { font-size: 60%; }
mjx-container [size="sm"] { font-size: 85%; }
mjx-container [size="lg"] { font-size: 120%; }
mjx-container [size="Lg"] { font-size: 144%; }
mjx-container [size="LG"] { font-size: 173%; }
mjx-container [size="hg"] { font-size: 207%; }
mjx-container [size="HG"] { font-size: 249%; }
mjx-container [width="full"] { width: 100%; }
mjx-box { display: inline-block; }
mjx-block { display: block; }
mjx-itable { display: inline-table; }
mjx-row { display: table-row; }
mjx-row > * { display: table-cell; }
mjx-mtext { display: inline-block; }
mjx-mstyle { display: inline-block; }
mjx-merror { display: inline-block; color: red; background-color: yellow; }
mjx-mphantom { visibility: hidden; }
mjx-assistive-mml { top: 0px; left: 0px; clip: rect(1px, 1px, 1px, 1px); user-select: none; position: absolute !important; padding: 1px 0px 0px !important; border: 0px !important; display: block !important; width: auto !important; overflow: hidden !important; }
mjx-assistive-mml[display="block"] { width: 100% !important; }
mjx-c::before { display: block; width: 0px; }
.MJX-TEX { font-family: MJXZERO, MJXTEX; }
.TEX-B { font-family: MJXZERO, MJXTEX-B; }
.TEX-I { font-family: MJXZERO, MJXTEX-I; }
.TEX-MI { font-family: MJXZERO, MJXTEX-MI; }
.TEX-BI { font-family: MJXZERO, MJXTEX-BI; }
.TEX-S1 { font-family: MJXZERO, MJXTEX-S1; }
.TEX-S2 { font-family: MJXZERO, MJXTEX-S2; }
.TEX-S3 { font-family: MJXZERO, MJXTEX-S3; }
.TEX-S4 { font-family: MJXZERO, MJXTEX-S4; }
.TEX-A { font-family: MJXZERO, MJXTEX-A; }
.TEX-C { font-family: MJXZERO, MJXTEX-C; }
.TEX-CB { font-family: MJXZERO, MJXTEX-CB; }
.TEX-FR { font-family: MJXZERO, MJXTEX-FR; }
.TEX-FRB { font-family: MJXZERO, MJXTEX-FRB; }
.TEX-SS { font-family: MJXZERO, MJXTEX-SS; }
.TEX-SSB { font-family: MJXZERO, MJXTEX-SSB; }
.TEX-SSI { font-family: MJXZERO, MJXTEX-SSI; }
.TEX-SC { font-family: MJXZERO, MJXTEX-SC; }
.TEX-T { font-family: MJXZERO, MJXTEX-T; }
.TEX-V { font-family: MJXZERO, MJXTEX-V; }
.TEX-VB { font-family: MJXZERO, MJXTEX-VB; }
mjx-stretchy-v mjx-c, mjx-stretchy-h mjx-c { font-family: MJXZERO, MJXTEX-S1, MJXTEX-S4, MJXTEX, MJXTEX-A !important; }
@font-face { font-family: MJXZERO; src: url("lib/fonts/mathjax_zero.woff") format("woff"); }
@font-face { font-family: MJXTEX; src: url("lib/fonts/mathjax_main-regular.woff") format("woff"); }
@font-face { font-family: MJXTEX-B; src: url("lib/fonts/mathjax_main-bold.woff") format("woff"); }
@font-face { font-family: MJXTEX-I; src: url("lib/fonts/mathjax_math-italic.woff") format("woff"); }
@font-face { font-family: MJXTEX-MI; src: url("lib/fonts/mathjax_main-italic.woff") format("woff"); }
@font-face { font-family: MJXTEX-BI; src: url("lib/fonts/mathjax_math-bolditalic.woff") format("woff"); }
@font-face { font-family: MJXTEX-S1; src: url("lib/fonts/mathjax_size1-regular.woff") format("woff"); }
@font-face { font-family: MJXTEX-S2; src: url("lib/fonts/mathjax_size2-regular.woff") format("woff"); }
@font-face { font-family: MJXTEX-S3; src: url("lib/fonts/mathjax_size3-regular.woff") format("woff"); }
@font-face { font-family: MJXTEX-S4; src: url("lib/fonts/mathjax_size4-regular.woff") format("woff"); }
@font-face { font-family: MJXTEX-A; src: url("lib/fonts/mathjax_ams-regular.woff") format("woff"); }
@font-face { font-family: MJXTEX-C; src: url("lib/fonts/mathjax_calligraphic-regular.woff") format("woff"); }
@font-face { font-family: MJXTEX-CB; src: url("lib/fonts/mathjax_calligraphic-bold.woff") format("woff"); }
@font-face { font-family: MJXTEX-FR; src: url("lib/fonts/mathjax_fraktur-regular.woff") format("woff"); }
@font-face { font-family: MJXTEX-FRB; src: url("lib/fonts/mathjax_fraktur-bold.woff") format("woff"); }
@font-face { font-family: MJXTEX-SS; src: url("lib/fonts/mathjax_sansserif-regular.woff") format("woff"); }
@font-face { font-family: MJXTEX-SSB; src: url("lib/fonts/mathjax_sansserif-bold.woff") format("woff"); }
@font-face { font-family: MJXTEX-SSI; src: url("lib/fonts/mathjax_sansserif-italic.woff") format("woff"); }
@font-face { font-family: MJXTEX-SC; src: url("lib/fonts/mathjax_script-regular.woff") format("woff"); }
@font-face { font-family: MJXTEX-T; src: url("lib/fonts/mathjax_typewriter-regular.woff") format("woff"); }
@font-face { font-family: MJXTEX-V; src: url("lib/fonts/mathjax_vector-regular.woff") format("woff"); }
@font-face { font-family: MJXTEX-VB; src: url("lib/fonts/mathjax_vector-bold.woff") format("woff"); }
</style><pre class="frontmatter language-yaml" tabindex="0" style="display: none;"><code class="language-yaml is-loaded"><span class="token key atrule">date</span><span class="token punctuation">:</span> <span class="token datetime number">2024-12-14</span>
<span class="token key atrule">title</span><span class="token punctuation">:</span> Bioinfo<span class="token punctuation">-</span>Matrix<span class="token punctuation">-</span>Operations
<span class="token key atrule">status</span><span class="token punctuation">:</span> DONE
<span class="token key atrule">author</span><span class="token punctuation">:</span>
  <span class="token punctuation">-</span> AllenYGY
<span class="token key atrule">tags</span><span class="token punctuation">:</span>
  <span class="token punctuation">-</span> NOTE</code><button class="copy-code-button"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon lucide-copy"><rect x="8" y="8" width="14" height="14" rx="2" ry="2"></rect><path d="M4 16c-1.1 0-2-.9-2-2V4c0-1.1.9-2 2-2h10c1.1 0 2 .9 2 2"></path></svg></button></pre><div class="markdown-preview-sizer markdown-preview-section"><h1 class="page-title heading inline-title" id="Bioinfo_Matrix_Operations"><p dir="auto">Bioinfo Matrix Operations</p></h1><div class="el-h1 heading-wrapper"><div class="heading-children"><div class="el-hr"><hr></div><div class="el-h2 is-collapsed heading-wrapper"><h2 data-heading="Matrix Factorization" dir="auto" class="heading" id="Matrix_Factorization"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Matrix Factorization</h2><div class="heading-children"><div class="el-p"><p dir="auto">Non-negative dataset</p></div><div class="el-ul"><ul>
<li data-line="0" dir="auto">Counters: 计数</li>
<li data-line="1" dir="auto">Quantities: 数量数据</li>
<li data-line="2" dir="auto">Intensities: 强度数据</li>
</ul></div><div class="el-h3 is-collapsed heading-wrapper"><h3 data-heading="NMF" dir="auto" class="heading" id="NMF"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>NMF</h3><div class="heading-children"><div class="el-p"><p dir="auto">NMF is used in a compositional model.<br>
Data are assumed to be non-negative </p></div><div class="el-p"><p dir="auto">Every data vector is explained as a purely constructive linear composition of a set of bases </p></div><div class="el-ul"><ul>
<li data-line="0" dir="auto"><span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D449 TEX-I"></mjx-c></mjx-mi><mjx-mo class="mjx-n" space="4"><mjx-c class="mjx-c3D"></mjx-c></mjx-mo><mjx-munder space="4" limits="false"><mjx-mo class="mjx-sop"><mjx-c class="mjx-c2211 TEX-S1"></mjx-c></mjx-mo><mjx-script style="vertical-align: -0.285em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-munder><mjx-msub space="2"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D464 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-msub><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D435 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-msub></mjx-math></mjx-container></span></li>
<li data-line="1" dir="auto">The bases <span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D435 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-mi class="mjx-i" size="s"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi></mjx-script></mjx-msub></mjx-math></mjx-container></span> are in the same domain as the data </li>
</ul></div><div class="el-p"><p dir="auto">Constructive composition: no subtraction allowed</p></div><div class="el-ul"><ul>
<li data-line="0" dir="auto">Weights <span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D464 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-mi class="mjx-i" size="s"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi></mjx-script></mjx-msub></mjx-math></mjx-container></span> must be non-negative</li>
<li data-line="1" dir="auto">All components of bases <span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D435 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-mi class="mjx-i" size="s"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi></mjx-script></mjx-msub></mjx-math></mjx-container></span> must be non-negative</li>
</ul></div><div class="el-p"><p dir="auto"><strong>Data Does Not Need to Be Non-Negative</strong>:</p></div><div class="el-ul"><ul>
<li data-line="0" dir="auto">Unlike traditional NMF where data is assumed to be non-negative, this generalization allows data to exist in <strong>any quadrant</strong>, including negative regions.</li>
</ul></div><div class="el-p"><p dir="auto"><strong>Non-Negativity Constraint</strong>:</p></div><div class="el-ul"><ul>
<li data-line="0" dir="auto">
<p><strong>Only the weights must remain non-negative</strong>, regardless of the data’s distribution.</p>
</li>
<li data-line="2" dir="auto">
<p>The model preserves the fundamental principle of constructive addition without subtracting components.</p>
</li>
</ul></div><div class="el-h4 heading-wrapper"><h4 data-heading="Minimizing Divergence" dir="auto" class="heading" id="Minimizing_Divergence"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Minimizing Divergence</h4><div class="heading-children"><div class="el-ul"><ul>
<li data-line="0" dir="auto"><span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D43F TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-mn class="mjx-n" size="s"><mjx-c class="mjx-c32"></mjx-c></mjx-mn></mjx-script></mjx-msub></mjx-math></mjx-container></span> Divergence: <span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D437 TEX-I"></mjx-c></mjx-mi><mjx-mo class="mjx-n"><mjx-c class="mjx-c28"></mjx-c></mjx-mo><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D449 TEX-I"></mjx-c></mjx-mi><mjx-mo class="mjx-n"><mjx-c class="mjx-c2C"></mjx-c></mjx-mo><mjx-mi class="mjx-i" space="2"><mjx-c class="mjx-c1D435 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D44A TEX-I"></mjx-c></mjx-mi><mjx-mo class="mjx-n"><mjx-c class="mjx-c29"></mjx-c></mjx-mo><mjx-mo class="mjx-n" space="4"><mjx-c class="mjx-c3D"></mjx-c></mjx-mo><mjx-munder space="4" limits="false"><mjx-mo class="mjx-sop"><mjx-c class="mjx-c2211 TEX-S1"></mjx-c></mjx-mo><mjx-script style="vertical-align: -0.285em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-munder><mjx-munder space="2" limits="false"><mjx-mo class="mjx-sop"><mjx-c class="mjx-c2211 TEX-S1"></mjx-c></mjx-mo><mjx-script style="vertical-align: -0.285em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D457 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-munder><mjx-mo class="mjx-n"><mjx-c class="mjx-c28"></mjx-c></mjx-mo><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D449 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em; margin-left: -0.186em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D457 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-msub><mjx-mo class="mjx-n" space="3"><mjx-c class="mjx-c2212"></mjx-c></mjx-mo><mjx-mi class="mjx-i" space="3"><mjx-c class="mjx-c1D435 TEX-I"></mjx-c></mjx-mi><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D44A TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em; margin-left: -0.104em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D457 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-msub><mjx-msup><mjx-mo class="mjx-n"><mjx-c class="mjx-c29"></mjx-c></mjx-mo><mjx-script style="vertical-align: 0.363em;"><mjx-mn class="mjx-n" size="s"><mjx-c class="mjx-c32"></mjx-c></mjx-mn></mjx-script></mjx-msup></mjx-math></mjx-container></span></li>
<li data-line="1" dir="auto"><span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D43E TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D43F TEX-I"></mjx-c></mjx-mi></mjx-math></mjx-container></span> Divergence: <span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-munder limits="false"><mjx-mo class="mjx-sop"><mjx-c class="mjx-c2211 TEX-S1"></mjx-c></mjx-mo><mjx-script style="vertical-align: -0.285em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-munder><mjx-munder space="2" limits="false"><mjx-mo class="mjx-sop"><mjx-c class="mjx-c2211 TEX-S1"></mjx-c></mjx-mo><mjx-script style="vertical-align: -0.285em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D457 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-munder><mjx-mo class="mjx-n"><mjx-c class="mjx-c28"></mjx-c></mjx-mo><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D449 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em; margin-left: -0.186em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D457 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D459 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D45C TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D454 TEX-I"></mjx-c></mjx-mi><mjx-mfrac><mjx-frac><mjx-num><mjx-nstrut></mjx-nstrut><mjx-msub size="s"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D449 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em; margin-left: -0.186em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D457 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-msub></mjx-num><mjx-dbox><mjx-dtable><mjx-line></mjx-line><mjx-row><mjx-den><mjx-dstrut></mjx-dstrut><mjx-mrow size="s"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D435 TEX-I"></mjx-c></mjx-mi><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D44A TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em; margin-left: -0.104em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D457 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-msub></mjx-mrow></mjx-den></mjx-row></mjx-dtable></mjx-dbox></mjx-frac></mjx-mfrac><mjx-mo class="mjx-n" space="3"><mjx-c class="mjx-c2B"></mjx-c></mjx-mo><mjx-msub space="3"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D449 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em; margin-left: -0.186em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D457 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-msub><mjx-mo class="mjx-n" space="3"><mjx-c class="mjx-c2212"></mjx-c></mjx-mo><mjx-mi class="mjx-i" space="3"><mjx-c class="mjx-c1D435 TEX-I"></mjx-c></mjx-mi><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D44A TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em; margin-left: -0.104em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D457 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-msub><mjx-mo class="mjx-n"><mjx-c class="mjx-c29"></mjx-c></mjx-mo></mjx-math></mjx-container></span></li>
</ul></div><div class="el-p"><p dir="auto">Iterative solution: </p></div><div class="el-ul"><ul>
<li data-line="0" dir="auto"><span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D435 TEX-I"></mjx-c></mjx-mi><mjx-mo class="mjx-n" space="4"><mjx-c class="mjx-c3D"></mjx-c></mjx-mo><mjx-mo class="mjx-n" space="4"><mjx-c class="mjx-c5B"></mjx-c></mjx-mo><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D449 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D443 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D45B TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D463 TEX-I"></mjx-c></mjx-mi><mjx-mo class="mjx-n"><mjx-c class="mjx-c28"></mjx-c></mjx-mo><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D44A TEX-I"></mjx-c></mjx-mi><mjx-msub><mjx-mo class="mjx-n"><mjx-c class="mjx-c29"></mjx-c></mjx-mo><mjx-script style="vertical-align: -0.15em;"><mjx-texatom size="s" texclass="ORD"><mjx-mo class="mjx-n"><mjx-c class="mjx-c2B"></mjx-c></mjx-mo></mjx-texatom></mjx-script></mjx-msub><mjx-mo class="mjx-n"><mjx-c class="mjx-c5D"></mjx-c></mjx-mo></mjx-math></mjx-container></span></li>
<li data-line="1" dir="auto"><span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D44A TEX-I"></mjx-c></mjx-mi><mjx-mo class="mjx-n" space="4"><mjx-c class="mjx-c3D"></mjx-c></mjx-mo><mjx-mo class="mjx-n" space="4"><mjx-c class="mjx-c5B"></mjx-c></mjx-mo><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D443 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D45B TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D463 TEX-I"></mjx-c></mjx-mi><mjx-mo class="mjx-n"><mjx-c class="mjx-c28"></mjx-c></mjx-mo><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D435 TEX-I"></mjx-c></mjx-mi><mjx-mo class="mjx-n"><mjx-c class="mjx-c29"></mjx-c></mjx-mo><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D449 TEX-I"></mjx-c></mjx-mi><mjx-msub><mjx-mo class="mjx-n"><mjx-c class="mjx-c5D"></mjx-c></mjx-mo><mjx-script style="vertical-align: -0.15em;"><mjx-texatom size="s" texclass="ORD"><mjx-mo class="mjx-n"><mjx-c class="mjx-c2B"></mjx-c></mjx-mo></mjx-texatom></mjx-script></mjx-msub></mjx-math></mjx-container></span></li>
<li data-line="2" dir="auto">Subscript <span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-mo class="mjx-n"><mjx-c class="mjx-c2B"></mjx-c></mjx-mo></mjx-math></mjx-container></span> indicates thresholding -values to 0.表示对负值进行阈值处理，将所有负值设置为 0，确保非负性。</li>
</ul></div><div class="el-p"><p dir="auto">NMF is not unique </p></div><div class="el-p"><p dir="auto">NMF is not hierarchical <strong>（非负矩阵分解并非分层结构）</strong></p></div><div class="el-ol"><ol>
<li data-line="0" dir="auto">Rank-1 NMF</li>
</ol></div><div class="el-ul"><ul>
<li data-line="0" dir="auto">
<p><strong>Rank-1 分解</strong>：将数据矩阵分解为单一基向量和权重向量的乘积。</p>
</li>
<li data-line="2" dir="auto">
<p>左侧的原始矩阵被近似分解为右侧的<strong>基矩阵</strong>和<strong>权重矩阵</strong>。</p>
</li>
<li data-line="4" dir="auto">
<p>这种低秩分解可以很好地捕捉数据的主成分，但可能忽略细节。</p>
</li>
</ul></div><div class="el-ol"><ol start="2">
<li data-line="0" dir="auto">Rank-2 NMF</li>
</ol></div><div class="el-ul"><ul>
<li data-line="0" dir="auto">
<p><strong>Rank-2 分解</strong>：将数据矩阵分解为两个基向量和对应权重的组合。</p>
</li>
<li data-line="2" dir="auto">
<p>分解为两个矩阵之和：</p>
</li>
<li data-line="4" dir="auto">
<p>数据矩阵通过两个基成分加权求和表示。</p>
</li>
<li data-line="6" dir="auto">
<p>Rank-2 分解能够捕捉更多的结构信息，但分解结果与 Rank-1 不具有分层关系。</p>
</li>
</ul></div><div class="el-p"><p dir="auto"><strong>非分层结构</strong>：</p></div><div class="el-ul"><ul>
<li data-line="0" dir="auto"><strong>增加秩</strong> 并不会简单地在前一级结果的基础上扩展，而是需要<strong>重新计算</strong>分解。</li>
</ul></div><div class="el-p"><p dir="auto">秩的选择影响分解的质量、稀疏性和可解释性。</p></div><div class="el-ul"><ul>
<li data-line="0" dir="auto"><strong>稀疏性（Sparsity）</strong>：较低秩分解更稀疏，较高秩分解捕捉更多细节。</li>
<li data-line="1" dir="auto"><strong>可解释性（Interpretability）</strong>：秩的选择会影响基向量的可解释性。</li>
<li data-line="2" dir="auto"><strong>统计保真度（Statistical Fidelity）</strong>：较高秩通常能够更好地重构数据，但可能过拟合。</li>
<li data-line="3" dir="auto"><strong>最佳秩</strong>&nbsp; k&nbsp; <strong>的选择</strong>： 并无统一的理论方法，需要通过实验来确定最优秩。</li>
</ul></div></div></div></div></div><div class="el-h3 is-collapsed heading-wrapper"><h3 data-heading="NMF General framework" dir="auto" class="heading" id="NMF_General_framework"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>NMF General framework</h3><div class="heading-children"><div class="el-p"><p dir="auto"><strong><em>Gradient descent generally slow</em></strong></p></div><div class="el-p"><p dir="auto"><strong><em>Stochastic gradient descent inappropriate</em></strong></p></div><div class="el-p"><p dir="auto"><strong><em>Key Approach: Alternating Minimization</em></strong></p></div><div class="el-ol"><ol>
<li data-line="0" dir="auto">Pick starting point <strong>L₀</strong> and <strong>R₀</strong>  </li>
<li data-line="1" dir="auto"><strong>while</strong> not converged <strong>do</strong><br>
3. Keep <strong>R</strong> fixed, optimize <strong>L</strong><br>
4. Keep <strong>L</strong> fixed, optimize <strong>R</strong><br>
5. <strong>end while</strong></li>
</ol></div><div class="el-p"><p dir="auto"><strong><em>Update steps 3 and 4 are easier than solving the full problem.</em></strong></p></div><div class="el-p"><p dir="auto"><strong><em>This approach is also called alternating projections or (block) coordinate descent.</em></strong></p></div><div class="el-p"><p dir="auto"><strong><em>Starting Point Options</em></strong></p></div><div class="el-ul"><ul>
<li data-line="0" dir="auto"><strong>Random</strong></li>
<li data-line="1" dir="auto"><strong>Multi-start initialization</strong>: Try multiple random starting points, run a few epochs, continue with the best.</li>
<li data-line="2" dir="auto"><strong>Based on SVD</strong></li>
<li data-line="3" dir="auto"><strong>...</strong></li>
</ul></div></div></div><div class="el-h3 heading-wrapper"><h3 data-heading="NMF in Bioinformatics" dir="auto" class="heading" id="NMF_in_Bioinformatics"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>NMF in Bioinformatics</h3><div class="heading-children"><div class="el-h4 heading-wrapper"><h4 data-heading="Example - 1" dir="auto" class="heading" id="Example_-_1"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Example - 1</h4><div class="heading-children"><div class="el-p"><p dir="auto"><strong>非负矩阵分解（NMF）</strong> 来识别癌症基因组数据中的<strong>多维模块</strong>，通过综合分析多平台基因组数据实现： <a data-tooltip-position="top" aria-label="https://academic.oup.com/nar/article/40/19/9379/2414808" rel="noopener nofollow" class="external-link" href="https://academic.oup.com/nar/article/40/19/9379/2414808" target="_blank">link</a></p></div><div class="el-ul"><ul>
<li data-line="0" dir="auto"><strong>DNA 甲基化（DM）</strong></li>
<li data-line="1" dir="auto"><strong>miRNA 表达（ME）</strong></li>
<li data-line="2" dir="auto"><strong>基因表达（GE）</strong></li>
</ul></div><div class="el-p"><p dir="auto"><strong>The Joint NMF Framework for Integrative Analysis</strong></p></div></div></div><div class="el-h4 heading-wrapper"><h4 data-heading="Example - 2" dir="auto" class="heading" id="Example_-_2"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Example - 2</h4><div class="heading-children"><div class="el-p"><p dir="auto">本示例展示了如何使用 <strong>种子非负矩阵分解（Seeded NMF）</strong> 来解析<strong>空间转录组学</strong>（Spatial Transcriptomics, ST）数据。</p></div><div class="el-ul"><ul>
<li data-line="0" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>空间转录组学（ST）技术</strong>：
<ul>
<li data-line="1" dir="auto">生成基因表达图谱，同时保留组织的空间上下文。</li>
<li data-line="2" dir="auto">每个空间点（spots）通常捕获多个细胞，缺乏<strong>单细胞分辨率</strong>。</li>
</ul>
</li>
<li data-line="3" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>目标</strong>：
<ul>
<li data-line="4" dir="auto">通过集成单细胞 RNA 测序（scRNA-seq）数据和 ST 数据，使用 NMF 进行<strong>去卷积（Deconvolution）</strong>。</li>
<li data-line="5" dir="auto">解释每个空间点的<strong>细胞组成</strong>。</li>
</ul>
</li>
</ul></div><div class="el-h5 heading-wrapper"><h5 data-heading="SPOTlight" dir="auto" class="heading" id="SPOTlight"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>SPOTlight</h5><div class="heading-children"><div class="el-p"><p dir="auto"><strong>SPOTlight 使用种子 NMF 回归（Seeded NMF Regression）</strong> 来实现目标：</p></div><div class="el-ol"><ol>
<li data-line="0" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>
<p><strong>学习单细胞数据中的主题特征（signatures）</strong>：</p>
<ul>
<li data-line="2" dir="auto">使用单细胞 RNA 测序数据，识别不同细胞类型的<strong>基因表达特征</strong>。</li>
</ul>
</li>
<li data-line="3" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>
<p><strong>解析空间数据</strong>：</p>
<ul>
<li data-line="5" dir="auto">将这些特征应用于 ST 数据，找到每个空间点的<strong>最佳加权细胞类型组合</strong>。</li>
</ul>
</li>
</ol></div></div></div><div class="el-h5 heading-wrapper"><h5 data-heading="NMF 分解过程" dir="auto" class="heading" id="NMF_分解过程"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>NMF 分解过程</h5><div class="heading-children"><div class="el-ul"><ul>
<li data-line="0" dir="auto">
<p><strong>初始化 (NMF Initialization)</strong></p>
</li>
<li data-line="2" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>
<p>数据分解为： <span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D449 TEX-I"></mjx-c></mjx-mi><mjx-mo class="mjx-n" space="4"><mjx-c class="mjx-c2248"></mjx-c></mjx-mo><mjx-mi class="mjx-i" space="4"><mjx-c class="mjx-c1D44A TEX-I"></mjx-c></mjx-mi><mjx-mo class="mjx-n" space="3"><mjx-c class="mjx-cD7"></mjx-c></mjx-mo><mjx-mi class="mjx-i" space="3"><mjx-c class="mjx-c1D43B TEX-I"></mjx-c></mjx-mi></mjx-math></mjx-container></span></p>
<ul>
<li data-line="3" dir="auto"><strong>W（基矩阵）</strong>：基因 × 主题（gene × topic）。</li>
<li data-line="4" dir="auto"><strong>H（系数矩阵）</strong>：主题 × 细胞（topic × cell）。</li>
<li data-line="5" dir="auto"><strong>V</strong>：基因 × 细胞的原始表达数据矩阵。</li>
</ul>
</li>
<li data-line="7" dir="auto">
<p><strong>分解步骤 (Factorization)</strong></p>
</li>
<li data-line="9" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>
<p>将 ST 数据 V′V' 分解为基矩阵 WW 和系数矩阵 HH：</p>
<ul>
<li data-line="10" dir="auto">WW：基因 × 主题矩阵（与单细胞特征关联）。</li>
<li data-line="11" dir="auto">HH：主题 × 空间点的权重矩阵。</li>
</ul>
</li>
</ul></div></div></div><div class="el-h5 heading-wrapper"><h5 data-heading="解析结果：Spot 去卷积" dir="auto" class="heading" id="解析结果：Spot_去卷积"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>解析结果：Spot 去卷积</h5><div class="heading-children"><div class="el-ul"><ul>
<li data-line="0" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>NNLS（非负最小二乘法）</strong>：
<ul>
<li data-line="1" dir="auto">用于优化权重矩阵 HH，以解释每个空间点的<strong>细胞组成</strong>。</li>
</ul>
</li>
<li data-line="2" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>结果展示</strong>：
<ul>
<li data-line="3" dir="auto">每个空间点（Spot 1、Spot 2 等）分解为多个细胞类型的加权组合，显示在<strong>饼图</strong>中。</li>
<li data-line="4" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>例如：
<ul>
<li data-line="5" dir="auto">Spot 1: 17% 细胞类型 A，17% 细胞类型 B，33% 细胞类型 C 等。</li>
</ul>
</li>
</ul>
</li>
</ul></div><div class="el-ol"><ol>
<li data-line="0" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>种子 NMF 的优势</strong>：
<ul>
<li data-line="1" dir="auto">使用单细胞数据作为“种子”初始化 WW，提高结果的生物学可解释性。</li>
</ul>
</li>
<li data-line="2" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>空间数据去卷积</strong>：
<ul>
<li data-line="3" dir="auto">SPOTlight 将单细胞数据与空间数据结合，揭示每个空间点的<strong>细胞组成</strong>。</li>
</ul>
</li>
<li data-line="4" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>应用场景</strong>：
<ul>
<li data-line="5" dir="auto">理解组织中的空间细胞异质性，特别在癌症等研究中具有重要意义。</li>
</ul>
</li>
</ol></div><div class="el-ul"><ul>
<li data-line="0" dir="auto"><a data-tooltip-position="top" aria-label="https://academic.oup.com/nar/article/49/9/e50/6120341" rel="noopener nofollow" class="external-link" href="https://academic.oup.com/nar/article/49/9/e50/6120341" target="_blank">Paper</a></li>
<li data-line="1" dir="auto"><a data-tooltip-position="top" aria-label="https://github.com/MarcElosua/SPOTlight" rel="noopener nofollow" class="external-link" href="https://github.com/MarcElosua/SPOTlight" target="_blank">SPOTlight Code</a></li>
</ul></div><div class="el-p"><p dir="auto"><strong>总结</strong>：通过种子 NMF 回归方法，SPOTlight 有效解析空间转录组学数据，结合单细胞数据揭示组织的细胞组成。  </p></div></div></div></div></div><div class="el-h4 heading-wrapper"><h4 data-heading="Example - 3" dir="auto" class="heading" id="Example_-_3"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Example - 3</h4><div class="heading-children"><div class="el-p"><p dir="auto"><a data-tooltip-position="top" aria-label="https://www.nature.com/articles/s41586-020-1943-3" rel="noopener nofollow" class="external-link" href="https://www.nature.com/articles/s41586-020-1943-3" target="_blank">Paper</a></p></div><div class="el-h5 heading-wrapper"><h5 data-heading="背景：人类癌症中的突变特征" dir="auto" class="heading" id="背景：人类癌症中的突变特征"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>背景：人类癌症中的突变特征</h5><div class="heading-children"><div class="el-p"><p dir="auto"><strong>目标</strong>：通过非负矩阵分解（NMF）方法，揭示不同癌症样本中的<strong>突变特征</strong>（mutational signatures）。<br>
<strong>数据</strong>：</p></div><div class="el-ul"><ul>
<li data-line="0" dir="auto">横轴：<strong>癌症样本</strong>（不同的癌症类型）。</li>
<li data-line="1" dir="auto">纵轴：<strong>突变特征</strong>（SBS1、SBS2、SBS3 等）。</li>
<li data-line="2" dir="auto">点的大小与颜色：代表不同特征在特定癌症样本中的<strong>突变贡献程度</strong>。</li>
</ul></div></div></div><div class="el-h5 heading-wrapper"><h5 data-heading="主要内容解析" dir="auto" class="heading" id="主要内容解析"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>主要内容解析</h5><div class="heading-children"><div class="el-ol"><ol>
<li data-line="0" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>横轴（Cancer Samples）</strong>：
<ul>
<li data-line="1" dir="auto">不同的癌症类型（如乳腺癌、肺癌等）。</li>
<li data-line="2" dir="auto">每一列表示一个癌症样本集合。</li>
</ul>
</li>
<li data-line="3" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>纵轴（Mutation Signatures）</strong>：
<ul>
<li data-line="4" dir="auto"><strong>SBS（Single Base Substitution）特征</strong>：代表单碱基替换的不同突变模式（SBS1、SBS2 等）。</li>
<li data-line="5" dir="auto">每一行代表一种特定的突变特征。</li>
</ul>
</li>
<li data-line="6" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>点的表示</strong>：
<ul>
<li data-line="7" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>颜色</strong>：不同颜色表示特征的突变机制，例如：
<ul>
<li data-line="8" dir="auto">橙色：APOBEC 活性</li>
<li data-line="9" dir="auto">紫色：DNA 修复缺陷</li>
<li data-line="10" dir="auto">蓝色：化学暴露或辐射</li>
</ul>
</li>
<li data-line="11" dir="auto"><strong>点的大小</strong>：表示特定突变特征在癌症样本中的<strong>贡献比例</strong>。</li>
<li data-line="12" dir="auto"><strong>颜色深浅</strong>：表明突变数量的多少，越深表示突变数量越高。</li>
</ul>
</li>
<li data-line="13" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>右侧图例（Proposed Aetiology）</strong>：
<ul>
<li data-line="14" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>列出了每个突变特征背后的<strong>潜在机制</strong>，例如：
<ul>
<li data-line="15" dir="auto"><strong>SBS1</strong>：5-甲基胞嘧啶脱氨</li>
<li data-line="16" dir="auto"><strong>SBS3</strong>：BRCA1/2 突变导致的 DNA 修复缺陷</li>
<li data-line="17" dir="auto"><strong>SBS4</strong>：与烟草吸烟相关</li>
</ul>
</li>
</ul>
</li>
</ol></div></div></div><div class="el-h5 heading-wrapper"><h5 data-heading="NMF 的应用" dir="auto" class="heading" id="NMF_的应用"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>NMF 的应用</h5><div class="heading-children"><div class="el-ul"><ul>
<li data-line="0" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>非负矩阵分解（NMF）</strong>：
<ul>
<li data-line="1" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>将突变数据矩阵分解为：<span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D44B TEX-I"></mjx-c></mjx-mi><mjx-mo class="mjx-n" space="4"><mjx-c class="mjx-c2248"></mjx-c></mjx-mo><mjx-mi class="mjx-i" space="4"><mjx-c class="mjx-c1D44A TEX-I"></mjx-c></mjx-mi><mjx-mo class="mjx-n" space="3"><mjx-c class="mjx-cD7"></mjx-c></mjx-mo><mjx-mi class="mjx-i" space="3"><mjx-c class="mjx-c1D43B TEX-I"></mjx-c></mjx-mi></mjx-math></mjx-container></span>
<ul>
<li data-line="2" dir="auto">X：突变数据矩阵（癌症样本 × 突变特征）。</li>
<li data-line="3" dir="auto">W：基因组突变的<strong>特征矩阵</strong>（表示不同特征的突变机制）。</li>
<li data-line="4" dir="auto">H：样本中的权重矩阵（表示各特征在不同样本中的贡献）。</li>
</ul>
</li>
<li data-line="5" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>NMF 的作用</strong>：
<ul>
<li data-line="6" dir="auto">提取癌症样本中的<strong>突变特征</strong>。</li>
<li data-line="7" dir="auto">识别不同癌症类型中的<strong>主要突变机制</strong>。</li>
</ul>
</li>
</ul>
</li>
</ul></div></div></div><div class="el-h5 heading-wrapper"><h5 data-heading="关键结果" dir="auto" class="heading" id="关键结果"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>关键结果</h5><div class="heading-children"><div class="el-ol"><ol>
<li data-line="0" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>
<p><strong>突变特征（SBS）与癌症类型的关联</strong>：</p>
<ul>
<li data-line="2" dir="auto">某些突变特征（如 SBS3）与<strong>DNA 修复缺陷</strong>相关（例如 BRCA1/2 突变）。</li>
<li data-line="3" dir="auto">SBS4 明显与<strong>烟草吸烟</strong>相关。</li>
<li data-line="4" dir="auto">特定癌症类型的突变模式与特定环境因素或基因缺陷有关。</li>
</ul>
</li>
<li data-line="5" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>
<p><strong>数据可视化</strong>：</p>
<ul>
<li data-line="7" dir="auto">点图展示了不同癌症样本中各突变特征的贡献大小。</li>
<li data-line="8" dir="auto">可直观地识别出哪些突变机制在某些癌症类型中占主导地位。</li>
</ul>
</li>
<li data-line="9" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>
<p><strong>突变机制</strong>：</p>
<ul>
<li data-line="11" dir="auto">提供了癌症突变的潜在生物学机制（如化学暴露、DNA 修复缺陷等），对癌症的病因研究具有重要意义。</li>
</ul>
</li>
</ol></div></div></div><div class="el-h5 heading-wrapper"><h5 data-heading="总结" dir="auto" class="heading" id="总结"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>总结</h5><div class="heading-children"><div class="el-p"><p dir="auto">通过 NMF 分解：</p></div><div class="el-ol"><ol>
<li data-line="0" dir="auto"><strong>揭示了人类癌症中的突变特征</strong>及其贡献程度。</li>
<li data-line="1" dir="auto">帮助识别不同癌症类型中的<strong>主要突变机制</strong>（例如化学暴露、DNA 修复缺陷等）。</li>
<li data-line="2" dir="auto">为癌症研究提供了一个整合、系统化的框架，揭示突变数据中的<strong>潜在模式</strong>和<strong>生物学机制</strong>。</li>
</ol></div></div></div></div></div></div></div></div></div><div class="el-h2 is-collapsed heading-wrapper"><h2 data-heading="Matrix Imputation" dir="auto" class="heading" id="Matrix_Imputation"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Matrix Imputation</h2><div class="heading-children"><div class="el-ul"><ul>
<li data-line="0" dir="auto">Proteomics </li>
<li data-line="1" dir="auto">Single cell RNA-seq </li>
<li data-line="2" dir="auto">Spatial Transcriptomics</li>
</ul></div><div class="el-h3 heading-wrapper"><h3 data-heading="Proteomics" dir="auto" class="heading" id="Proteomics"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Proteomics</h3><div class="heading-children"><div class="el-p"><p dir="auto">在蛋白质组学（Proteomics）数据分析中，缺失值是常见问题，解决缺失值的主要方法可以分为以下几类：</p></div><div class="el-h4 heading-wrapper"><h4 data-heading="Single Value Replacement" dir="auto" class="heading" id="Single_Value_Replacement"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Single Value Replacement</h4><div class="heading-children"><div class="el-p"><p dir="auto">单值替换</p></div><div class="el-ul"><ul>
<li data-line="0" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>半最小值法（Half Minimum）</strong> 
<ul>
<li data-line="1" dir="auto">将缺失值替换为所有观测数据中最小值的一半。</li>
<li data-line="2" dir="auto"><strong>适用场景</strong>：MNAR（Missing Not At Random，非随机缺失）。</li>
</ul>
</li>
<li data-line="3" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>均值替换（Mean）</strong>
<ul>
<li data-line="4" dir="auto">将缺失值替换为所有观测值的平均值。</li>
<li data-line="5" dir="auto"><strong>适用场景</strong>：MCAR/MAR（随机缺失/完全随机缺失）。</li>
</ul>
</li>
</ul></div></div></div><div class="el-h4 heading-wrapper"><h4 data-heading="Global Structure Low-Rank Matrix Factorization" dir="auto" class="heading" id="Global_Structure_Low-Rank_Matrix_Factorization"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Global Structure Low-Rank Matrix Factorization</h4><div class="heading-children"><div class="el-p"><p dir="auto">全局结构低秩矩阵分解</p></div><div class="el-ul"><ul>
<li data-line="0" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>PPCA（概率主成分分析）
<ul>
<li data-line="1" dir="auto">基于低秩的概率主成分分析方法插补缺失值。</li>
<li data-line="2" dir="auto"><strong>适用场景</strong>：MCAR/MAR。</li>
</ul>
</li>
<li data-line="3" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>NIPALS（非线性迭代部分最小二乘法）
<ul>
<li data-line="4" dir="auto">通过低秩矩阵分解迭代估算缺失值。</li>
<li data-line="5" dir="auto"><strong>适用场景</strong>：MCAR/MAR。</li>
</ul>
</li>
<li data-line="6" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>SVD（奇异值分解）
<ul>
<li data-line="7" dir="auto">通过低秩 SVD 矩阵分解进行插补，利用数据的全局结构。</li>
<li data-line="8" dir="auto"><strong>适用场景</strong>：MCAR/MAR。</li>
</ul>
</li>
<li data-line="9" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>SVT（奇异值阈值化）
<ul>
<li data-line="10" dir="auto">通过核范数最小化方法解决缺失值问题，适用于低秩数据。</li>
<li data-line="11" dir="auto"><strong>适用场景</strong>：MCAR/MAR。</li>
</ul>
</li>
</ul></div></div></div><div class="el-h4 heading-wrapper"><h4 data-heading="Local Similarity" dir="auto" class="heading" id="Local_Similarity"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Local Similarity</h4><div class="heading-children"><div class="el-p"><p dir="auto">局部相似性</p></div><div class="el-ul"><ul>
<li data-line="0" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>样本最近邻（Sample-wise KNN）</strong>
<ul>
<li data-line="1" dir="auto">基于样本间的局部相似性，通过 kk-最近邻样本的加权平均值插补缺失值。</li>
<li data-line="2" dir="auto"><strong>适用场景</strong>：MAR（随机缺失）。</li>
</ul>
</li>
<li data-line="3" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>蛋白质最近邻（Protein-wise KNN）</strong>
<ul>
<li data-line="4" dir="auto">基于蛋白质之间的相似性，通过 kk-最近邻蛋白质的加权平均插补缺失值。</li>
<li data-line="5" dir="auto"><strong>适用场景</strong>：MAR（随机缺失）。</li>
</ul>
</li>
</ul></div><div class="el-ol"><ol>
<li data-line="0" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>MNAR: 非随机缺失 
<ul>
<li data-line="1" dir="auto">缺失值与未观测的数据本身有关，例如“半最小值法”。</li>
</ul>
</li>
<li data-line="2" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>MCAR: 完全随机缺失
<ul>
<li data-line="3" dir="auto">缺失值与观测数据无关，例如 PPCA、SVD、SVT 等矩阵分解方法。</li>
</ul>
</li>
<li data-line="4" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>MAR: 随机缺失
<ul>
<li data-line="5" dir="auto">缺失值与观测数据有关，但与未观测数据无关，例如 KNN 方法。</li>
</ul>
</li>
</ol></div></div></div></div></div><div class="el-h3 heading-wrapper"><h3 data-heading="Single cell RNA-seq" dir="auto" class="heading" id="Single_cell_RNA-seq"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Single cell RNA-seq</h3><div class="heading-children"><div class="el-ul"><ul>
<li data-line="0" dir="auto">MAGIC</li>
<li data-line="1" dir="auto">scImpute</li>
<li data-line="2" dir="auto">SAVER</li>
<li data-line="3" dir="auto">ALRA</li>
<li data-line="4" dir="auto">SAUCIE</li>
</ul></div><div class="el-h4 heading-wrapper"><h4 data-heading="MAGIC" dir="auto" class="heading" id="MAGIC"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>MAGIC</h4><div class="heading-children"><div class="el-h5 heading-wrapper"><h5 data-heading="1. 原始数据 (Original Data)" dir="auto" class="heading" id="1._原始数据_(Original_Data)"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>1. 原始数据 (Original Data)</h5><div class="heading-children"><div class="el-ul"><ul>
<li data-line="0" dir="auto">输入是单细胞转录组的原始基因表达矩阵。</li>
<li data-line="1" dir="auto">数据特点：由于技术限制，数据通常稀疏且有噪声。</li>
<li data-line="2" dir="auto">可视化：点图展示了细胞在一个隐空间中的分布。</li>
</ul></div></div></div><div class="el-h5 heading-wrapper"><h5 data-heading="2. 计算距离矩阵 (Calculate Distances)" dir="auto" class="heading" id="2._计算距离矩阵_(Calculate_Distances)"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>2. 计算距离矩阵 (Calculate Distances)</h5><div class="heading-children"><div class="el-ul"><ul>
<li data-line="0" dir="auto"><strong>目的</strong>：计算细胞间的距离，基于基因表达值。</li>
<li data-line="1" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>方法</strong>：
<ul>
<li data-line="2" dir="auto">使用欧几里得距离或其他距离度量。</li>
</ul>
</li>
<li data-line="3" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>结果</strong>：
<ul>
<li data-line="4" dir="auto">生成一个对称的距离矩阵，表示细胞两两之间的相似性。</li>
<li data-line="5" dir="auto">点图进一步展示细胞在空间中的连接趋势。</li>
</ul>
</li>
</ul></div></div></div><div class="el-h5 heading-wrapper"><h5 data-heading="3. 计算亲和矩阵 (Calculate Affinities)" dir="auto" class="heading" id="3._计算亲和矩阵_(Calculate_Affinities)"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>3. 计算亲和矩阵 (Calculate Affinities)</h5><div class="heading-children"><div class="el-ul"><ul>
<li data-line="0" dir="auto"><strong>目的</strong>：将距离矩阵转化为亲和度矩阵（Affinity Matrix）。</li>
<li data-line="1" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>方法</strong>：
<ul>
<li data-line="2" dir="auto">使用高斯核（Gaussian Kernel），将距离映射为亲和度。</li>
<li data-line="3" dir="auto">公式：<span class="math math-block is-loaded"><mjx-container class="MathJax" jax="CHTML" display="true"><mjx-math display="true" class="MJX-TEX" style="margin-left: 0px; margin-right: 0px;"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D434 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D453 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D453 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D45B TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D461 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D466 TEX-I"></mjx-c></mjx-mi><mjx-mtext class="mjx-n"><mjx-c class="mjx-cA0"></mjx-c></mjx-mtext><mjx-mo class="mjx-n" space="4"><mjx-c class="mjx-c3D"></mjx-c></mjx-mo><mjx-mi class="mjx-i" space="4"><mjx-c class="mjx-c1D452 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D465 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D45D TEX-I"></mjx-c></mjx-mi><mjx-mo class="mjx-n"><mjx-c class="mjx-c28"></mjx-c></mjx-mo><mjx-mo class="mjx-n"><mjx-c class="mjx-c2212"></mjx-c></mjx-mo><mjx-mfrac><mjx-frac type="d"><mjx-num><mjx-nstrut type="d"></mjx-nstrut><mjx-mrow><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D437 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D460 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D461 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D44E TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D45B TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D450 TEX-I"></mjx-c></mjx-mi><mjx-msup><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D452 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: 0.363em;"><mjx-texatom size="s" texclass="ORD"><mjx-mn class="mjx-n"><mjx-c class="mjx-c32"></mjx-c></mjx-mn></mjx-texatom></mjx-script></mjx-msup></mjx-mrow></mjx-num><mjx-dbox><mjx-dtable><mjx-line type="d"></mjx-line><mjx-row><mjx-den><mjx-dstrut type="d"></mjx-dstrut><mjx-mrow><mjx-mn class="mjx-n"><mjx-c class="mjx-c32"></mjx-c></mjx-mn><mjx-msup><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D70E TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: 0.289em;"><mjx-texatom size="s" texclass="ORD"><mjx-mn class="mjx-n"><mjx-c class="mjx-c32"></mjx-c></mjx-mn></mjx-texatom></mjx-script></mjx-msup></mjx-mrow></mjx-den></mjx-row></mjx-dtable></mjx-dbox></mjx-frac></mjx-mfrac><mjx-mo class="mjx-n"><mjx-c class="mjx-c29"></mjx-c></mjx-mo></mjx-math></mjx-container></span></li>
</ul>
</li>
<li data-line="4" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>结果</strong>：
<ul>
<li data-line="5" dir="auto">亲和度矩阵中，较小的距离对应高的亲和值，较大的距离对应低的亲和值。</li>
</ul>
</li>
</ul></div></div></div><div class="el-h5 heading-wrapper"><h5 data-heading="4. 马尔科夫标准化 (Markov Normalization)" dir="auto" class="heading" id="4._马尔科夫标准化_(Markov_Normalization)"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>4. 马尔科夫标准化 (Markov Normalization)</h5><div class="heading-children"><div class="el-ul"><ul>
<li data-line="0" dir="auto"><strong>目的</strong>：对亲和度矩阵进行标准化，使其满足马尔科夫性质（每行和为1）。</li>
<li data-line="1" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>方法</strong>：
<ul>
<li data-line="2" dir="auto">对矩阵进行归一化，得到马尔科夫过渡矩阵。</li>
</ul>
</li>
<li data-line="3" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>结果</strong>：
<ul>
<li data-line="4" dir="auto">生成的矩阵可用于模拟细胞间的扩散过程。</li>
</ul>
</li>
</ul></div></div></div><div class="el-h5 heading-wrapper"><h5 data-heading="5. 马尔科夫矩阵指数化 (Exponentiate Markov Matrix)" dir="auto" class="heading" id="5._马尔科夫矩阵指数化_(Exponentiate_Markov_Matrix)"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>5. 马尔科夫矩阵指数化 (Exponentiate Markov Matrix)</h5><div class="heading-children"><div class="el-ul"><ul>
<li data-line="0" dir="auto"><strong>目的</strong>：模拟扩散过程，扩散步数由参数 tt 决定。</li>
<li data-line="1" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>方法</strong>：
<ul>
<li data-line="2" dir="auto">对马尔科夫矩阵进行 <span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D461 TEX-I"></mjx-c></mjx-mi></mjx-math></mjx-container></span> 次幂 <span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-msup><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D440 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: 0.363em; margin-left: 0.054em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D461 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-msup></mjx-math></mjx-container></span></li>
</ul>
</li>
<li data-line="3" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>效果</strong>：
<ul>
<li data-line="4" dir="auto">增加扩散步数 <span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D461 TEX-I"></mjx-c></mjx-mi></mjx-math></mjx-container></span> 可以加深数据的平滑程度，消除噪声。</li>
</ul>
</li>
</ul></div></div></div><div class="el-h5 heading-wrapper"><h5 data-heading="6. 基因表达补全 (Impute Gene Expression)" dir="auto" class="heading" id="6._基因表达补全_(Impute_Gene_Expression)"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>6. 基因表达补全 (Impute Gene Expression)</h5><div class="heading-children"><div class="el-ul"><ul>
<li data-line="0" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>计算</strong>：
<ul>
<li data-line="1" dir="auto">将扩散后的马尔科夫矩阵 <span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-msup><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D440 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: 0.363em; margin-left: 0.054em;"><mjx-mi class="mjx-i" size="s"><mjx-c class="mjx-c1D461 TEX-I"></mjx-c></mjx-mi></mjx-script></mjx-msup></mjx-math></mjx-container></span> 应用于原始数据矩阵 <span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D437 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-utext variant="italic" style="font-family: MJXZERO, serif; font-style: italic;">：</mjx-utext></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D437 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D45A TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D45D TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D462 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D461 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D452 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D451 TEX-I"></mjx-c></mjx-mi><mjx-mo class="mjx-n" space="4"><mjx-c class="mjx-c3D"></mjx-c></mjx-mo><mjx-mi class="mjx-i" space="4"><mjx-c class="mjx-c1D440 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D461 TEX-I"></mjx-c></mjx-mi><mjx-mo class="mjx-n" space="3"><mjx-c class="mjx-c22C5"></mjx-c></mjx-mo><mjx-mi class="mjx-i" space="3"><mjx-c class="mjx-c1D437 TEX-I"></mjx-c></mjx-mi><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D437 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-texatom size="s" texclass="ORD"><mjx-mtext class="mjx-n"><mjx-c class="mjx-c69"></mjx-c><mjx-c class="mjx-c6D"></mjx-c><mjx-c class="mjx-c70"></mjx-c><mjx-c class="mjx-c75"></mjx-c><mjx-c class="mjx-c74"></mjx-c><mjx-c class="mjx-c65"></mjx-c><mjx-c class="mjx-c64"></mjx-c></mjx-mtext></mjx-texatom></mjx-script></mjx-msub><mjx-mo class="mjx-n" space="4"><mjx-c class="mjx-c3D"></mjx-c></mjx-mo><mjx-msup space="4"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D440 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: 0.363em; margin-left: 0.054em;"><mjx-mi class="mjx-i" size="s"><mjx-c class="mjx-c1D461 TEX-I"></mjx-c></mjx-mi></mjx-script></mjx-msup><mjx-mo class="mjx-n" space="3"><mjx-c class="mjx-c22C5"></mjx-c></mjx-mo><mjx-mi class="mjx-i" space="3"><mjx-c class="mjx-c1D437 TEX-I"></mjx-c></mjx-mi></mjx-math></mjx-container></span></li>
<li data-line="2" dir="auto">对补全后的数据进行重缩放。</li>
</ul>
</li>
<li data-line="3" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>结果</strong>：
<ul>
<li data-line="4" dir="auto">得到一个平滑且补全的基因表达矩阵，能够更好地反映生物学信号。</li>
</ul>
</li>
</ul></div><div class="el-p"><p dir="auto">MAGIC 算法通过以下步骤实现数据补全：</p></div><div class="el-ol"><ol>
<li data-line="0" dir="auto">计算距离。</li>
<li data-line="1" dir="auto">转化为亲和度。</li>
<li data-line="2" dir="auto">使用马尔科夫模型模拟数据扩散。</li>
<li data-line="3" dir="auto">对基因表达进行平滑和补全。</li>
</ol></div><div class="el-p"><p dir="auto">这是一种有效的基于图的方法，可显著改善单细胞数据的解析质量。</p></div></div></div></div></div></div></div><div class="el-h3 heading-wrapper"><h3 data-heading="Spatial Transcriptomics" dir="auto" class="heading" id="Spatial_Transcriptomics"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Spatial Transcriptomics</h3><div class="heading-children"><div class="el-ul"><ul>
<li data-line="0" dir="auto">stDiff</li>
<li data-line="1" dir="auto">GNTD</li>
</ul></div><div class="el-h4 heading-wrapper"><h4 data-heading="stDiff" dir="auto" class="heading" id="stDiff"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>stDiff</h4><div class="heading-children"></div></div></div></div></div></div><div class="el-h2 heading-wrapper"><h2 data-heading="Matrix Dimension Reduction" dir="auto" class="heading" id="Matrix_Dimension_Reduction"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Matrix Dimension Reduction</h2><div class="heading-children"><div class="el-h3 heading-wrapper"><h3 data-heading="Linear Methods" dir="auto" class="heading" id="Linear_Methods"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Linear Methods</h3><div class="heading-children"><div class="el-h4 heading-wrapper"><h4 data-heading="PCA" dir="auto" class="heading" id="PCA"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>PCA</h4><div class="heading-children"></div></div><div class="el-h4 heading-wrapper"><h4 data-heading="cMDS" dir="auto" class="heading" id="cMDS"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>cMDS</h4><div class="heading-children"></div></div></div></div><div class="el-h3 heading-wrapper"><h3 data-heading="Non-linear methods" dir="auto" class="heading" id="Non-linear_methods"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Non-linear methods</h3><div class="heading-children"><div class="el-ul"><ul>
<li data-line="0" dir="auto">t-SNE</li>
<li data-line="1" dir="auto">UMAP</li>
<li data-line="2" dir="auto">KPCA</li>
<li data-line="3" dir="auto">mMDS</li>
<li data-line="4" dir="auto">Isomap </li>
<li data-line="5" dir="auto">LLE (Locally Linear Embedding)</li>
<li data-line="6" dir="auto">Laplacian Eigenmap</li>
</ul></div><div class="el-h4 heading-wrapper"><h4 data-heading="t-SNE" dir="auto" class="heading" id="t-SNE"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>t-SNE</h4><div class="heading-children"><div class="el-p"><p dir="auto">t-SNE（t-Distributed Stochastic Neighbor Embedding）是一种非线性降维技术，广泛用于高维数据的可视化。由 Laurens van der Maaten 和 Geoffrey Hinton 于 2008 年提出，t-SNE 能很好地保留数据的局部结构，因此特别适合于聚类或高维数据的可视化任务。</p></div><div class="el-h5 heading-wrapper"><h5 data-heading="t-SNE 的基本原理" dir="auto" class="heading" id="t-SNE_的基本原理"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>t-SNE 的基本原理</h5><div class="heading-children"><div class="el-p"><p dir="auto"><strong>高维空间的相似度计算</strong>：</p></div><div class="el-ul"><ul>
<li data-line="0" dir="auto">
<p>在高维空间中，t-SNE 用 <strong>高斯分布</strong> 表示点之间的相似性。</p>
<div class="math math-block is-loaded"><mjx-container class="MathJax" jax="CHTML" display="true"><mjx-math display="true" class="MJX-TEX" style="margin-left: 0px; margin-right: 0px;"><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D45D TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D457 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-msub><mjx-mo class="mjx-n" space="4"><mjx-c class="mjx-c3D"></mjx-c></mjx-mo><mjx-mfrac space="4"><mjx-frac type="d"><mjx-num><mjx-nstrut type="d"></mjx-nstrut><mjx-mfrac><mjx-frac><mjx-num><mjx-nstrut></mjx-nstrut><mjx-mrow size="s"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D452 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D465 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D45D TEX-I"></mjx-c></mjx-mi><mjx-mo class="mjx-n"><mjx-c class="mjx-c28"></mjx-c></mjx-mo><mjx-mo class="mjx-n"><mjx-c class="mjx-c2212"></mjx-c></mjx-mo><mjx-mo class="mjx-n"><mjx-c class="mjx-c7C"></mjx-c></mjx-mo><mjx-mo class="mjx-n"><mjx-c class="mjx-c7C"></mjx-c></mjx-mo><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D465 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-mi class="mjx-i" size="s"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi></mjx-script></mjx-msub><mjx-mo class="mjx-n"><mjx-c class="mjx-c2212"></mjx-c></mjx-mo><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D465 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-mi class="mjx-i" size="s"><mjx-c class="mjx-c1D457 TEX-I"></mjx-c></mjx-mi></mjx-script></mjx-msub><mjx-mo class="mjx-n"><mjx-c class="mjx-c7C"></mjx-c></mjx-mo><mjx-msup><mjx-mo class="mjx-n"><mjx-c class="mjx-c7C"></mjx-c></mjx-mo><mjx-script style="vertical-align: 0.363em;"><mjx-texatom size="s" texclass="ORD"><mjx-mn class="mjx-n"><mjx-c class="mjx-c32"></mjx-c></mjx-mn></mjx-texatom></mjx-script></mjx-msup><mjx-mo class="mjx-n"><mjx-c class="mjx-c29"></mjx-c></mjx-mo></mjx-mrow></mjx-num><mjx-dbox><mjx-dtable><mjx-line></mjx-line><mjx-row><mjx-den><mjx-dstrut></mjx-dstrut><mjx-mrow size="s"><mjx-mn class="mjx-n"><mjx-c class="mjx-c32"></mjx-c></mjx-mn><mjx-msubsup><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D70E TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.294em; margin-left: 0px;"><mjx-texatom size="s" texclass="ORD"><mjx-mn class="mjx-n"><mjx-c class="mjx-c32"></mjx-c></mjx-mn></mjx-texatom><mjx-spacer style="margin-top: 0.18em;"></mjx-spacer><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-msubsup></mjx-mrow></mjx-den></mjx-row></mjx-dtable></mjx-dbox></mjx-frac></mjx-mfrac></mjx-num><mjx-dbox><mjx-dtable><mjx-line type="d"></mjx-line><mjx-row><mjx-den><mjx-dstrut type="d"></mjx-dstrut><mjx-mrow><mjx-texatom texclass="ORD"><mjx-munder limits="false"><mjx-mo class="mjx-sop"><mjx-c class="mjx-c2211 TEX-S1"></mjx-c></mjx-mo><mjx-script style="vertical-align: -0.285em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D458 TEX-I"></mjx-c></mjx-mi><mjx-mo class="mjx-n"><mjx-c class="mjx-c2260"></mjx-c></mjx-mo><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-munder><mjx-mi class="mjx-i" space="2"><mjx-c class="mjx-c1D452 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D465 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D45D TEX-I"></mjx-c></mjx-mi><mjx-mo class="mjx-n"><mjx-c class="mjx-c28"></mjx-c></mjx-mo><mjx-mo class="mjx-n"><mjx-c class="mjx-c2212"></mjx-c></mjx-mo><mjx-mo class="mjx-n"><mjx-c class="mjx-c7C"></mjx-c></mjx-mo><mjx-mo class="mjx-n"><mjx-c class="mjx-c7C"></mjx-c></mjx-mo><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D465 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-mi class="mjx-i" size="s"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi></mjx-script></mjx-msub><mjx-mo class="mjx-n" space="3"><mjx-c class="mjx-c2212"></mjx-c></mjx-mo><mjx-msub space="3"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D465 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-mi class="mjx-i" size="s"><mjx-c class="mjx-c1D458 TEX-I"></mjx-c></mjx-mi></mjx-script></mjx-msub><mjx-mo class="mjx-n"><mjx-c class="mjx-c7C"></mjx-c></mjx-mo><mjx-msup><mjx-mo class="mjx-n"><mjx-c class="mjx-c7C"></mjx-c></mjx-mo><mjx-script style="vertical-align: 0.289em;"><mjx-texatom size="s" texclass="ORD"><mjx-mn class="mjx-n"><mjx-c class="mjx-c32"></mjx-c></mjx-mn></mjx-texatom></mjx-script></mjx-msup><mjx-mo class="mjx-n"><mjx-c class="mjx-c29"></mjx-c></mjx-mo></mjx-texatom><mjx-texatom texclass="ORD"><mjx-mn class="mjx-n"><mjx-c class="mjx-c32"></mjx-c></mjx-mn><mjx-msubsup><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D70E TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.294em; margin-left: 0px;"><mjx-texatom size="s" texclass="ORD"><mjx-mn class="mjx-n"><mjx-c class="mjx-c32"></mjx-c></mjx-mn></mjx-texatom><mjx-spacer style="margin-top: 0.18em;"></mjx-spacer><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-msubsup></mjx-texatom></mjx-mrow></mjx-den></mjx-row></mjx-dtable></mjx-dbox></mjx-frac></mjx-mfrac></mjx-math></mjx-container></div>
</li>
<li data-line="5" dir="auto">
<p>其中，<span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D70E TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-mi class="mjx-i" size="s"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi></mjx-script></mjx-msub></mjx-math></mjx-container></span> 是点 <span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi></mjx-math></mjx-container></span> 的局部尺度参数，通过<strong>预定义的困惑度（Perplexity）</strong> 调整。</p>
</li>
</ul></div><div class="el-p"><p dir="auto"><strong>低维空间的相似度计算</strong>：</p></div><div class="el-ul"><ul>
<li data-line="0" dir="auto">在低维空间中，点之间的相似性 <span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D45E TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em; margin-left: -0.014em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D457 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-msub></mjx-math></mjx-container></span> 用 <strong>学生分布（t-distribution）</strong> 表示： </li>
</ul></div><div class="el-div"><div class="math math-block is-loaded"><mjx-container class="MathJax" jax="CHTML" display="true"><mjx-math display="true" class="MJX-TEX" style="margin-left: 0px; margin-right: 0px;"><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D45E TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em; margin-left: -0.014em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D457 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-msub><mjx-mo class="mjx-n" space="4"><mjx-c class="mjx-c3D"></mjx-c></mjx-mo><mjx-mfrac space="4"><mjx-frac type="d"><mjx-num><mjx-nstrut type="d"></mjx-nstrut><mjx-mrow><mjx-mo class="mjx-n"><mjx-c class="mjx-c28"></mjx-c></mjx-mo><mjx-mn class="mjx-n"><mjx-c class="mjx-c31"></mjx-c></mjx-mn><mjx-mo class="mjx-n" space="3"><mjx-c class="mjx-c2B"></mjx-c></mjx-mo><mjx-mo class="mjx-n" space="3"><mjx-c class="mjx-c7C"></mjx-c></mjx-mo><mjx-mo class="mjx-n"><mjx-c class="mjx-c7C"></mjx-c></mjx-mo><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D466 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-mi class="mjx-i" size="s"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi></mjx-script></mjx-msub><mjx-mo class="mjx-n" space="3"><mjx-c class="mjx-c2212"></mjx-c></mjx-mo><mjx-msub space="3"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D466 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-mi class="mjx-i" size="s"><mjx-c class="mjx-c1D457 TEX-I"></mjx-c></mjx-mi></mjx-script></mjx-msub><mjx-mo class="mjx-n"><mjx-c class="mjx-c7C"></mjx-c></mjx-mo><mjx-msup><mjx-mo class="mjx-n"><mjx-c class="mjx-c7C"></mjx-c></mjx-mo><mjx-script style="vertical-align: 0.363em;"><mjx-texatom size="s" texclass="ORD"><mjx-mn class="mjx-n"><mjx-c class="mjx-c32"></mjx-c></mjx-mn></mjx-texatom></mjx-script></mjx-msup><mjx-msup><mjx-mo class="mjx-n"><mjx-c class="mjx-c29"></mjx-c></mjx-mo><mjx-script style="vertical-align: 0.363em;"><mjx-texatom size="s" texclass="ORD"><mjx-mo class="mjx-n"><mjx-c class="mjx-c2212"></mjx-c></mjx-mo><mjx-mn class="mjx-n"><mjx-c class="mjx-c31"></mjx-c></mjx-mn></mjx-texatom></mjx-script></mjx-msup></mjx-mrow></mjx-num><mjx-dbox><mjx-dtable><mjx-line type="d"></mjx-line><mjx-row><mjx-den><mjx-dstrut type="d"></mjx-dstrut><mjx-mrow><mjx-munder limits="false"><mjx-mo class="mjx-sop"><mjx-c class="mjx-c2211 TEX-S1"></mjx-c></mjx-mo><mjx-script style="vertical-align: -0.285em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D458 TEX-I"></mjx-c></mjx-mi><mjx-mo class="mjx-n"><mjx-c class="mjx-c2260"></mjx-c></mjx-mo><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D459 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-munder><mjx-mo class="mjx-n"><mjx-c class="mjx-c28"></mjx-c></mjx-mo><mjx-mn class="mjx-n"><mjx-c class="mjx-c31"></mjx-c></mjx-mn><mjx-mo class="mjx-n" space="3"><mjx-c class="mjx-c2B"></mjx-c></mjx-mo><mjx-mo class="mjx-n" space="3"><mjx-c class="mjx-c7C"></mjx-c></mjx-mo><mjx-mo class="mjx-n"><mjx-c class="mjx-c7C"></mjx-c></mjx-mo><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D466 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-mi class="mjx-i" size="s"><mjx-c class="mjx-c1D458 TEX-I"></mjx-c></mjx-mi></mjx-script></mjx-msub><mjx-mo class="mjx-n" space="3"><mjx-c class="mjx-c2212"></mjx-c></mjx-mo><mjx-msub space="3"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D466 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-mi class="mjx-i" size="s"><mjx-c class="mjx-c1D459 TEX-I"></mjx-c></mjx-mi></mjx-script></mjx-msub><mjx-mo class="mjx-n"><mjx-c class="mjx-c7C"></mjx-c></mjx-mo><mjx-msup><mjx-mo class="mjx-n"><mjx-c class="mjx-c7C"></mjx-c></mjx-mo><mjx-script style="vertical-align: 0.289em;"><mjx-texatom size="s" texclass="ORD"><mjx-mn class="mjx-n"><mjx-c class="mjx-c32"></mjx-c></mjx-mn></mjx-texatom></mjx-script></mjx-msup><mjx-msup><mjx-mo class="mjx-n"><mjx-c class="mjx-c29"></mjx-c></mjx-mo><mjx-script style="vertical-align: 0.289em;"><mjx-texatom size="s" texclass="ORD"><mjx-mo class="mjx-n"><mjx-c class="mjx-c2212"></mjx-c></mjx-mo><mjx-mn class="mjx-n"><mjx-c class="mjx-c31"></mjx-c></mjx-mn></mjx-texatom></mjx-script></mjx-msup></mjx-mrow></mjx-den></mjx-row></mjx-dtable></mjx-dbox></mjx-frac></mjx-mfrac></mjx-math></mjx-container></div></div><div class="el-p"><p dir="auto">这种分布的长尾特性使得低维空间更容易保留数据的局部结构。</p></div><div class="el-p"><p dir="auto"><strong>优化目标</strong>：</p></div><div class="el-ul"><ul>
<li data-line="0" dir="auto">t-SNE 的目标是使高维空间的相似性 <span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D45D TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D457 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-msub></mjx-math></mjx-container></span> 和低维空间的相似性 <span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D45E TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em; margin-left: -0.014em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D457 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-msub></mjx-math></mjx-container></span> 尽可能接近。使用 Kullback-Leibler 散度（KL 散度）作为优化目标： </li>
</ul></div><div class="el-div"><div class="math math-block is-loaded"><mjx-container class="MathJax" jax="CHTML" display="true"><mjx-math display="true" class="MJX-TEX" style="margin-left: 0px; margin-right: 0px;"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D43E TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D43F TEX-I"></mjx-c></mjx-mi><mjx-mo class="mjx-n"><mjx-c class="mjx-c28"></mjx-c></mjx-mo><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D443 TEX-I"></mjx-c></mjx-mi><mjx-mo class="mjx-n"><mjx-c class="mjx-c7C"></mjx-c></mjx-mo><mjx-mo class="mjx-n"><mjx-c class="mjx-c7C"></mjx-c></mjx-mo><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D444 TEX-I"></mjx-c></mjx-mi><mjx-mo class="mjx-n"><mjx-c class="mjx-c29"></mjx-c></mjx-mo><mjx-mo class="mjx-n" space="4"><mjx-c class="mjx-c3D"></mjx-c></mjx-mo><mjx-munder space="4"><mjx-row><mjx-base><mjx-mo class="mjx-lop"><mjx-c class="mjx-c2211 TEX-S2"></mjx-c></mjx-mo></mjx-base></mjx-row><mjx-row><mjx-under style="padding-top: 0.167em; padding-left: 0.6em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-under></mjx-row></mjx-munder><mjx-munder space="2"><mjx-row><mjx-base><mjx-mo class="mjx-lop"><mjx-c class="mjx-c2211 TEX-S2"></mjx-c></mjx-mo></mjx-base></mjx-row><mjx-row><mjx-under style="padding-top: 0.167em; padding-left: 0.576em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D457 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-under></mjx-row></mjx-munder><mjx-msub space="2"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D45D TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D457 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D459 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D45C TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D454 TEX-I"></mjx-c></mjx-mi><mjx-mfrac><mjx-frac type="d"><mjx-num><mjx-nstrut type="d"></mjx-nstrut><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D45D TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D457 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-msub></mjx-num><mjx-dbox><mjx-dtable><mjx-line type="d"></mjx-line><mjx-row><mjx-den><mjx-dstrut type="d"></mjx-dstrut><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D45E TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em; margin-left: -0.014em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D457 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-msub></mjx-den></mjx-row></mjx-dtable></mjx-dbox></mjx-frac></mjx-mfrac></mjx-math></mjx-container></div></div><div class="el-ul"><ul>
<li data-line="0" dir="auto">通过梯度下降算法最小化这个损失函数。</li>
</ul></div><div class="el-p"><p dir="auto"><img alt="t-SNE" src="https://cdn.jsdelivr.net/gh/AllenYGY/ImageSpace@main/uPic/NCz6jn.png" referrerpolicy="no-referrer"></p></div></div></div><div class="el-h5 heading-wrapper"><h5 data-heading="t-SNE 的核心参数" dir="auto" class="heading" id="t-SNE_的核心参数"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>t-SNE 的核心参数</h5><div class="heading-children"><div class="el-p"><p dir="auto"><strong>困惑度（Perplexity）</strong>：</p></div><div class="el-ul"><ul>
<li data-line="0" dir="auto">用于平衡局部和全局结构的影响，通常值在 5 到 50 之间。</li>
<li data-line="1" dir="auto">困惑度越高，模型关注的数据范围越大。</li>
</ul></div><div class="el-p"><p dir="auto"><strong>学习率（Learning Rate）</strong>：</p></div><div class="el-ul"><ul>
<li data-line="0" dir="auto">对优化过程的速度和质量有较大影响。</li>
<li data-line="1" dir="auto">通常建议的值为 200，但可以根据数据调整。</li>
</ul></div><div class="el-p"><p dir="auto"><strong>迭代次数</strong>：</p></div><div class="el-ul"><ul>
<li data-line="0" dir="auto">t-SNE 通常需要 1000 次以上的迭代才能收敛。</li>
</ul></div></div></div><div class="el-h5 heading-wrapper"><h5 data-heading="优点：" dir="auto" class="heading" id="优点："><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>优点：</h5><div class="heading-children"><div class="el-ol"><ol>
<li data-line="0" dir="auto">非常适合用于高维数据的可视化。</li>
<li data-line="1" dir="auto">能很好地保留数据的局部结构，易于发现数据的聚类特性。</li>
</ol></div></div></div><div class="el-h5 heading-wrapper"><h5 data-heading="局限性：" dir="auto" class="heading" id="局限性："><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>局限性：</h5><div class="heading-children"><div class="el-ol"><ol>
<li data-line="0" dir="auto">计算复杂度较高，不适合特别大规模的数据集。</li>
<li data-line="1" dir="auto">不适合用作一般降维算法（例如 PCA 的替代）。</li>
<li data-line="2" dir="auto">超参数（如困惑度、学习率）对结果影响较大，需要仔细调整。</li>
<li data-line="3" dir="auto">结果的随机性强，运行多次可能得到不同的结果。</li>
</ol></div></div></div></div></div><div class="el-h4 heading-wrapper"><h4 data-heading="UMAP" dir="auto" class="heading" id="UMAP"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>UMAP</h4><div class="heading-children"><div class="el-p"><p dir="auto">t-SNE converts the distance between the two points into a probability in an ad hoc manner and attempts to preserve this probability </p></div><div class="el-p"><p dir="auto">UMAP, starts with a totally different idea: To construct a manifold from only a sampling of the points on it </p></div><div class="el-p"><p dir="auto"><img alt="UMAP" src="https://cdn.jsdelivr.net/gh/AllenYGY/ImageSpace@main/uPic/KxVWgb.png" referrerpolicy="no-referrer"></p></div><div class="el-h5 heading-wrapper"><h5 data-heading="UMAP 的基本原理" dir="auto" class="heading" id="UMAP_的基本原理"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>UMAP 的基本原理</h5><div class="heading-children"><div class="el-p"><p dir="auto">UMAP 的理论基础是流形假设（Manifold Hypothesis），即高维数据在本质上分布在低维的流形上。UMAP 通过以下步骤实现降维：</p></div></div></div><div class="el-h5 heading-wrapper"><h5 data-heading="构建高维空间的邻域图" dir="auto" class="heading" id="构建高维空间的邻域图"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>构建高维空间的邻域图</strong></h5><div class="heading-children"><div class="el-ul"><ul>
<li data-line="0" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>计算邻居：</strong> 使用 k 近邻算法（k-NN）在高维空间中找到每个点的最近邻。
<ul>
<li data-line="1" dir="auto">用户需设置 <code>n_neighbors</code> 参数，决定邻域大小。</li>
</ul>
</li>
<li data-line="2" dir="auto"><strong>定义邻居相似度：</strong> 使用高斯分布计算邻域内点之间的相似性。</li>
</ul></div><div class="el-div"><div class="math math-block is-loaded"><mjx-container class="MathJax" jax="CHTML" display="true"><mjx-math display="true" class="MJX-TEX" style="margin-left: 0px; margin-right: 0px;"><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D464 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D457 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-msub><mjx-mo class="mjx-n" space="4"><mjx-c class="mjx-c3D"></mjx-c></mjx-mo><mjx-mi class="mjx-i" space="4"><mjx-c class="mjx-c1D452 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D465 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D45D TEX-I"></mjx-c></mjx-mi><mjx-mo class="mjx-n"><mjx-c class="mjx-c28"></mjx-c></mjx-mo><mjx-mo class="mjx-n"><mjx-c class="mjx-c2212"></mjx-c></mjx-mo><mjx-mfrac><mjx-frac type="d"><mjx-num><mjx-nstrut type="d"></mjx-nstrut><mjx-mrow><mjx-mo class="mjx-n"><mjx-c class="mjx-c7C"></mjx-c></mjx-mo><mjx-mo class="mjx-n"><mjx-c class="mjx-c7C"></mjx-c></mjx-mo><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D465 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-mi class="mjx-i" size="s"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi></mjx-script></mjx-msub><mjx-mo class="mjx-n" space="3"><mjx-c class="mjx-c2212"></mjx-c></mjx-mo><mjx-msub space="3"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D465 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-mi class="mjx-i" size="s"><mjx-c class="mjx-c1D457 TEX-I"></mjx-c></mjx-mi></mjx-script></mjx-msub><mjx-mo class="mjx-n"><mjx-c class="mjx-c7C"></mjx-c></mjx-mo><mjx-mo class="mjx-n"><mjx-c class="mjx-c7C"></mjx-c></mjx-mo><mjx-mo class="mjx-n" space="3"><mjx-c class="mjx-c2212"></mjx-c></mjx-mo><mjx-msub space="3"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D70C TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-mi class="mjx-i" size="s"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi></mjx-script></mjx-msub></mjx-mrow></mjx-num><mjx-dbox><mjx-dtable><mjx-line type="d"></mjx-line><mjx-row><mjx-den><mjx-dstrut type="d"></mjx-dstrut><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D70E TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-mi class="mjx-i" size="s"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi></mjx-script></mjx-msub></mjx-den></mjx-row></mjx-dtable></mjx-dbox></mjx-frac></mjx-mfrac><mjx-mo class="mjx-n"><mjx-c class="mjx-c29"></mjx-c></mjx-mo></mjx-math></mjx-container></div></div><div class="el-p"><p dir="auto">其中：</p></div><div class="el-ul"><ul>
<li data-line="0" dir="auto"><span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D70C TEX-I"></mjx-c></mjx-mi></mjx-math></mjx-container></span>: 平滑参数，确保最近的邻居有较小的权重。</li>
<li data-line="1" dir="auto"><span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D70E TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-mi class="mjx-i" size="s"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi></mjx-script></mjx-msub></mjx-math></mjx-container></span>：局部尺度参数，通过优化确保邻域结构的一致性。</li>
<li data-line="4" dir="auto">生成一个加权无向图，点之间的权重代表相似性。</li>
</ul></div></div></div><div class="el-h5 heading-wrapper"><h5 data-heading="将高维邻域图嵌入到低维空间" dir="auto" class="heading" id="将高维邻域图嵌入到低维空间"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>将高维邻域图嵌入到低维空间</strong></h5><div class="heading-children"><div class="el-ul"><ul>
<li data-line="0" dir="auto">在低维空间中，点之间的距离用一种<strong>单调下降的曲线（通常是倒数函数）</strong>来衡量。 </li>
</ul></div><div class="el-div"><div class="math math-block is-loaded"><mjx-container class="MathJax" jax="CHTML" display="true"><mjx-math display="true" class="MJX-TEX" style="margin-left: 0px; margin-right: 0px;"><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D451 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-texatom size="s" texclass="ORD"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D457 TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-msub><mjx-mo class="mjx-n" space="4"><mjx-c class="mjx-c3D"></mjx-c></mjx-mo><mjx-mfrac space="4"><mjx-frac type="d"><mjx-num><mjx-nstrut type="d"></mjx-nstrut><mjx-mn class="mjx-n"><mjx-c class="mjx-c31"></mjx-c></mjx-mn></mjx-num><mjx-dbox><mjx-dtable><mjx-line type="d"></mjx-line><mjx-row><mjx-den><mjx-dstrut type="d"></mjx-dstrut><mjx-mrow><mjx-mn class="mjx-n"><mjx-c class="mjx-c31"></mjx-c></mjx-mn><mjx-mo class="mjx-n" space="3"><mjx-c class="mjx-c2B"></mjx-c></mjx-mo><mjx-mi class="mjx-i" space="3"><mjx-c class="mjx-c1D44E TEX-I"></mjx-c></mjx-mi><mjx-mo class="mjx-n"><mjx-c class="mjx-c7C"></mjx-c></mjx-mo><mjx-mo class="mjx-n"><mjx-c class="mjx-c7C"></mjx-c></mjx-mo><mjx-msub><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D466 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-mi class="mjx-i" size="s"><mjx-c class="mjx-c1D456 TEX-I"></mjx-c></mjx-mi></mjx-script></mjx-msub><mjx-mo class="mjx-n" space="3"><mjx-c class="mjx-c2212"></mjx-c></mjx-mo><mjx-msub space="3"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D466 TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: -0.15em;"><mjx-mi class="mjx-i" size="s"><mjx-c class="mjx-c1D457 TEX-I"></mjx-c></mjx-mi></mjx-script></mjx-msub><mjx-mo class="mjx-n"><mjx-c class="mjx-c7C"></mjx-c></mjx-mo><mjx-msup><mjx-mo class="mjx-n"><mjx-c class="mjx-c7C"></mjx-c></mjx-mo><mjx-script style="vertical-align: 0.289em;"><mjx-texatom size="s" texclass="ORD"><mjx-mn class="mjx-n"><mjx-c class="mjx-c32"></mjx-c></mjx-mn><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D44F TEX-I"></mjx-c></mjx-mi></mjx-texatom></mjx-script></mjx-msup></mjx-mrow></mjx-den></mjx-row></mjx-dtable></mjx-dbox></mjx-frac></mjx-mfrac></mjx-math></mjx-container></div></div><div class="el-p"><p dir="auto">其中 <span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D44E TEX-I"></mjx-c></mjx-mi></mjx-math></mjx-container></span> 和 <span class="math math-inline is-loaded"><mjx-container class="MathJax" jax="CHTML"><mjx-math class="MJX-TEX"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D44F TEX-I"></mjx-c></mjx-mi></mjx-math></mjx-container></span> 是超参数，通过优化选择，确保低维距离和高维相似性具有一致性。</p></div><div class="el-ul"><ul>
<li data-line="0" dir="auto">优化目标是最小化高维邻域图和低维空间图之间的差异，使用交叉熵损失函数进行迭代优化。</li>
</ul></div></div></div><div class="el-h5 heading-wrapper"><h5 data-heading="优化过程" dir="auto" class="heading" id="优化过程"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><strong>优化过程</strong></h5><div class="heading-children"><div class="el-ul"><ul>
<li data-line="0" dir="auto">UMAP 使用随机梯度下降（SGD）优化，将高维数据点映射到低维空间，同时保留高维数据的局部和全局结构。</li>
<li data-line="1" dir="auto">与 t-SNE 不同，UMAP 的优化更高效，适用于大规模数据。</li>
</ul></div></div></div><div class="el-h5 heading-wrapper"><h5 data-heading="UMAP 的核心参数" dir="auto" class="heading" id="UMAP_的核心参数"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>UMAP 的核心参数</h5><div class="heading-children"><div class="el-ol"><ol>
<li data-line="0" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>
<p><strong><code>n_neighbors</code>（邻居数量）</strong></p>
<ul>
<li data-line="2" dir="auto">控制每个点的局部邻域大小，影响全局和局部结构的平衡。</li>
<li data-line="3" dir="auto">小值：强调局部结构。</li>
<li data-line="4" dir="auto">大值：保留更多全局结构。</li>
</ul>
</li>
<li data-line="5" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>
<p><strong><code>min_dist</code>（最小距离）</strong></p>
<ul>
<li data-line="7" dir="auto">控制嵌入空间中点之间的最小距离，影响聚类结果的紧密程度。</li>
<li data-line="8" dir="auto">小值：点簇更紧凑，聚类更明显。</li>
<li data-line="9" dir="auto">大值：点分布更稀疏。</li>
</ul>
</li>
<li data-line="10" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>
<p><strong><code>n_components</code>（降维后的维度）</strong></p>
<ul>
<li data-line="12" dir="auto">默认值为 2，适合可视化，但也可以设置为更高维度，用于特征提取或其他任务。</li>
</ul>
</li>
<li data-line="13" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>
<p><strong><code>metric</code>（距离度量）</strong></p>
<ul>
<li data-line="15" dir="auto">用于衡量高维空间中的点对相似性。默认是欧几里得距离，也支持其他距离（如曼哈顿距离、余弦相似度等）。</li>
</ul>
</li>
</ol></div></div></div><div class="el-h5 heading-wrapper"><h5 data-heading="优点" dir="auto" class="heading" id="优点"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>优点</h5><div class="heading-children"><div class="el-ol"><ol>
<li data-line="0" dir="auto"><strong>速度快</strong>：UMAP 的计算效率比 t-SNE 高，适合大规模数据。</li>
<li data-line="1" dir="auto"><strong>全局结构保留更好</strong>：UMAP 在降维时能同时保留局部和全局结构。</li>
<li data-line="2" dir="auto"><strong>参数可调性强</strong>：可以通过 <code>n_neighbors</code> 和 <code>min_dist</code> 调整对局部或全局结构的关注程度。</li>
<li data-line="3" dir="auto"><strong>适用范围广</strong>：支持多种距离度量，可应用于不同类型的数据（如图像、文本、基因数据）。</li>
</ol></div></div></div><div class="el-h5 heading-wrapper"><h5 data-heading="局限性" dir="auto" class="heading" id="局限性"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>局限性</h5><div class="heading-children"><div class="el-ol"><ol>
<li data-line="0" dir="auto">参数较多，需要根据具体任务调整。</li>
<li data-line="1" dir="auto">对嵌入的随机性敏感，不同运行可能会有略微不同的结果。</li>
</ol></div></div></div></div></div></div></div><div class="el-h3 heading-wrapper"><h3 data-heading="UMAP 与 t-SNE 的比较" dir="auto" class="heading" id="UMAP_与_t-SNE_的比较"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>UMAP 与 t-SNE 的比较</h3><div class="heading-children"><div class="el-table" dir="ltr" style="overflow-x: auto;"><table>
<thead>
<tr>
<th dir="ltr">特性</th>
<th dir="ltr">UMAP</th>
<th dir="ltr">t-SNE</th>
</tr>
</thead>
<tbody>
<tr>
<td dir="ltr"><strong>速度</strong></td>
<td dir="ltr">更快，适合大规模数据</td>
<td dir="ltr">较慢，特别是大数据集时</td>
</tr>
<tr>
<td dir="ltr"><strong>全局结构</strong></td>
<td dir="ltr">更好地保留全局结构</td>
<td dir="ltr">更关注局部结构</td>
</tr>
<tr>
<td dir="ltr"><strong>参数调节</strong></td>
<td dir="ltr">更灵活，支持多种距离度量</td>
<td dir="ltr">参数少但对结果影响较大</td>
</tr>
<tr>
<td dir="ltr"><strong>嵌入稳定性</strong></td>
<td dir="ltr">嵌入结果更稳定</td>
<td dir="ltr">嵌入结果具有一定随机性</td>
</tr>
<tr>
<td dir="ltr"><strong>使用场景</strong></td>
<td dir="ltr">数据探索、聚类、降维</td>
<td dir="ltr">数据可视化（主要用于 2D 或 3D）</td>
</tr>
</tbody>
</table></div></div></div></div></div><div class="el-h2 heading-wrapper"><h2 data-heading="3D reconstruction from single cell RNA-seq" dir="auto" class="heading" id="3D_reconstruction_from_single_cell_RNA-seq"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>3D reconstruction from single cell RNA-seq</h2><div class="heading-children"><div class="mod-footer mod-ui"></div></div></div></div></div></div></div></div><div class="sidebar-right sidebar"><div class="sidebar-handle"></div><div class="sidebar-topbar"><div class="topbar-content"></div><div class="clickable-icon sidebar-collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="100%" height="100%" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon"><path d="M21 3H3C1.89543 3 1 3.89543 1 5V19C1 20.1046 1.89543 21 3 21H21C22.1046 21 23 20.1046 23 19V5C23 3.89543 22.1046 3 21 3Z"></path><path d="M10 4V20"></path><path d="M4 7H7"></path><path d="M4 10H7"></path><path d="M4 13H7"></path></svg></div></div><div class="sidebar-content"><div class="tree-container mod-root nav-folder tree-item outline-tree" data-depth="0"><div class="tree-header"><span class="sidebar-section-header">Table Of Contents</span><button class="clickable-icon collapse-tree-button" aria-label="Collapse All"><svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 20 20" fill="none" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"></svg></button></div><div class="tree-scroll-area tree-item-children nav-folder-children"><div class="tree-item mod-tree-folder nav-folder mod-collapsible is-collapsed" style="display: none;"></div><div class="tree-item" data-depth="1"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#Bioinfo_Matrix_Operations" data-path=""><div class="tree-item-contents heading-link" heading-name="Bioinfo Matrix Operations"><span class="tree-item-title">Bioinfo Matrix Operations</span></div></a><div class="tree-item-children nav-folder-children"><div class="tree-item mod-collapsible" data-depth="2"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#Matrix_Factorization" data-path=""><div class="tree-item-contents heading-link" heading-name="Matrix Factorization"><div class="collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><span class="tree-item-title">Matrix Factorization</span></div></a><div class="tree-item-children nav-folder-children"><div class="tree-item mod-collapsible" data-depth="3"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#NMF" data-path=""><div class="tree-item-contents heading-link" heading-name="NMF"><div class="collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><span class="tree-item-title">NMF</span></div></a><div class="tree-item-children nav-folder-children"><div class="tree-item" data-depth="4"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#Minimizing_Divergence" data-path=""><div class="tree-item-contents heading-link" heading-name="Minimizing Divergence"><span class="tree-item-title">Minimizing Divergence</span></div></a><div class="tree-item-children nav-folder-children"></div></div></div></div><div class="tree-item" data-depth="3"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#NMF_General_framework" data-path=""><div class="tree-item-contents heading-link" heading-name="NMF General framework"><span class="tree-item-title">NMF General framework</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item mod-collapsible" data-depth="3"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#NMF_in_Bioinformatics" data-path=""><div class="tree-item-contents heading-link" heading-name="NMF in Bioinformatics"><div class="collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><span class="tree-item-title">NMF in Bioinformatics</span></div></a><div class="tree-item-children nav-folder-children"><div class="tree-item" data-depth="4"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#Example_-_1" data-path=""><div class="tree-item-contents heading-link" heading-name="Example - 1"><span class="tree-item-title">Example - 1</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item mod-collapsible" data-depth="4"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#Example_-_2" data-path=""><div class="tree-item-contents heading-link" heading-name="Example - 2"><div class="collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><span class="tree-item-title">Example - 2</span></div></a><div class="tree-item-children nav-folder-children"><div class="tree-item" data-depth="5"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#SPOTlight" data-path=""><div class="tree-item-contents heading-link" heading-name="SPOTlight"><span class="tree-item-title">SPOTlight</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#NMF_分解过程" data-path=""><div class="tree-item-contents heading-link" heading-name="NMF 分解过程"><span class="tree-item-title">NMF 分解过程</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#解析结果：Spot_去卷积" data-path=""><div class="tree-item-contents heading-link" heading-name="解析结果：Spot 去卷积"><span class="tree-item-title">解析结果：Spot 去卷积</span></div></a><div class="tree-item-children nav-folder-children"></div></div></div></div><div class="tree-item mod-collapsible" data-depth="4"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#Example_-_3" data-path=""><div class="tree-item-contents heading-link" heading-name="Example - 3"><div class="collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><span class="tree-item-title">Example - 3</span></div></a><div class="tree-item-children nav-folder-children"><div class="tree-item" data-depth="5"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#背景：人类癌症中的突变特征" data-path=""><div class="tree-item-contents heading-link" heading-name="背景：人类癌症中的突变特征"><span class="tree-item-title">背景：人类癌症中的突变特征</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#主要内容解析" data-path=""><div class="tree-item-contents heading-link" heading-name="主要内容解析"><span class="tree-item-title">主要内容解析</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#NMF_的应用" data-path=""><div class="tree-item-contents heading-link" heading-name="NMF 的应用"><span class="tree-item-title">NMF 的应用</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#关键结果" data-path=""><div class="tree-item-contents heading-link" heading-name="关键结果"><span class="tree-item-title">关键结果</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#总结" data-path=""><div class="tree-item-contents heading-link" heading-name="总结"><span class="tree-item-title">总结</span></div></a><div class="tree-item-children nav-folder-children"></div></div></div></div></div></div></div></div><div class="tree-item mod-collapsible" data-depth="2"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#Matrix_Imputation" data-path=""><div class="tree-item-contents heading-link" heading-name="Matrix Imputation"><div class="collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><span class="tree-item-title">Matrix Imputation</span></div></a><div class="tree-item-children nav-folder-children"><div class="tree-item mod-collapsible" data-depth="3"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#Proteomics" data-path=""><div class="tree-item-contents heading-link" heading-name="Proteomics"><div class="collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><span class="tree-item-title">Proteomics</span></div></a><div class="tree-item-children nav-folder-children"><div class="tree-item" data-depth="4"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#Single_Value_Replacement" data-path=""><div class="tree-item-contents heading-link" heading-name="Single Value Replacement"><span class="tree-item-title">Single Value Replacement</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="4"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#Global_Structure_Low-Rank_Matrix_Factorization" data-path=""><div class="tree-item-contents heading-link" heading-name="Global Structure Low-Rank Matrix Factorization"><span class="tree-item-title">Global Structure Low-Rank Matrix Factorization</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="4"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#Local_Similarity" data-path=""><div class="tree-item-contents heading-link" heading-name="Local Similarity"><span class="tree-item-title">Local Similarity</span></div></a><div class="tree-item-children nav-folder-children"></div></div></div></div><div class="tree-item mod-collapsible" data-depth="3"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#Single_cell_RNA-seq" data-path=""><div class="tree-item-contents heading-link" heading-name="Single cell RNA-seq"><div class="collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><span class="tree-item-title">Single cell RNA-seq</span></div></a><div class="tree-item-children nav-folder-children"><div class="tree-item mod-collapsible" data-depth="4"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#MAGIC" data-path=""><div class="tree-item-contents heading-link" heading-name="MAGIC"><div class="collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><span class="tree-item-title">MAGIC</span></div></a><div class="tree-item-children nav-folder-children"><div class="tree-item" data-depth="5"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#1._原始数据_(Original_Data)" data-path=""><div class="tree-item-contents heading-link" heading-name="1. 原始数据 (Original Data)"><span class="tree-item-title">1. 
原始数据 (Original Data)
</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#2._计算距离矩阵_(Calculate_Distances)" data-path=""><div class="tree-item-contents heading-link" heading-name="2. 计算距离矩阵 (Calculate Distances)"><span class="tree-item-title">2. 
计算距离矩阵 (Calculate Distances)
</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#3._计算亲和矩阵_(Calculate_Affinities)" data-path=""><div class="tree-item-contents heading-link" heading-name="3. 计算亲和矩阵 (Calculate Affinities)"><span class="tree-item-title">3. 
计算亲和矩阵 (Calculate Affinities)
</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#4._马尔科夫标准化_(Markov_Normalization)" data-path=""><div class="tree-item-contents heading-link" heading-name="4. 马尔科夫标准化 (Markov Normalization)"><span class="tree-item-title">4. 
马尔科夫标准化 (Markov Normalization)
</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#5._马尔科夫矩阵指数化_(Exponentiate_Markov_Matrix)" data-path=""><div class="tree-item-contents heading-link" heading-name="5. 马尔科夫矩阵指数化 (Exponentiate Markov Matrix)"><span class="tree-item-title">5. 
马尔科夫矩阵指数化 (Exponentiate Markov Matrix)
</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#6._基因表达补全_(Impute_Gene_Expression)" data-path=""><div class="tree-item-contents heading-link" heading-name="6. 基因表达补全 (Impute Gene Expression)"><span class="tree-item-title">6. 
基因表达补全 (Impute Gene Expression)
</span></div></a><div class="tree-item-children nav-folder-children"></div></div></div></div></div></div><div class="tree-item mod-collapsible" data-depth="3"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#Spatial_Transcriptomics" data-path=""><div class="tree-item-contents heading-link" heading-name="Spatial Transcriptomics"><div class="collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><span class="tree-item-title">Spatial Transcriptomics</span></div></a><div class="tree-item-children nav-folder-children"><div class="tree-item" data-depth="4"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#stDiff" data-path=""><div class="tree-item-contents heading-link" heading-name="stDiff"><span class="tree-item-title">stDiff</span></div></a><div class="tree-item-children nav-folder-children"></div></div></div></div></div></div><div class="tree-item mod-collapsible" data-depth="2"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#Matrix_Dimension_Reduction" data-path=""><div class="tree-item-contents heading-link" heading-name="Matrix Dimension Reduction"><div class="collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><span class="tree-item-title">Matrix Dimension Reduction</span></div></a><div class="tree-item-children nav-folder-children"><div class="tree-item mod-collapsible" data-depth="3"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#Linear_Methods" data-path=""><div class="tree-item-contents heading-link" heading-name="Linear Methods"><div class="collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><span class="tree-item-title">Linear Methods</span></div></a><div class="tree-item-children nav-folder-children"><div class="tree-item" data-depth="4"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#PCA" data-path=""><div class="tree-item-contents heading-link" heading-name="PCA"><span class="tree-item-title">PCA</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="4"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#cMDS" data-path=""><div class="tree-item-contents heading-link" heading-name="cMDS"><span class="tree-item-title">cMDS</span></div></a><div class="tree-item-children nav-folder-children"></div></div></div></div><div class="tree-item mod-collapsible" data-depth="3"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#Non-linear_methods" data-path=""><div class="tree-item-contents heading-link" heading-name="Non-linear methods"><div class="collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><span class="tree-item-title">Non-linear methods</span></div></a><div class="tree-item-children nav-folder-children"><div class="tree-item mod-collapsible" data-depth="4"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#t-SNE" data-path=""><div class="tree-item-contents heading-link" heading-name="t-SNE"><div class="collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><span class="tree-item-title">t-SNE</span></div></a><div class="tree-item-children nav-folder-children"><div class="tree-item" data-depth="5"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#t-SNE_的基本原理" data-path=""><div class="tree-item-contents heading-link" heading-name="t-SNE 的基本原理"><span class="tree-item-title">t-SNE 的基本原理</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#t-SNE_的核心参数" data-path=""><div class="tree-item-contents heading-link" heading-name="t-SNE 的核心参数"><span class="tree-item-title">t-SNE 的核心参数</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#优点：" data-path=""><div class="tree-item-contents heading-link" heading-name="优点："><span class="tree-item-title">优点：</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#局限性：" data-path=""><div class="tree-item-contents heading-link" heading-name="局限性："><span class="tree-item-title">局限性：</span></div></a><div class="tree-item-children nav-folder-children"></div></div></div></div><div class="tree-item mod-collapsible" data-depth="4"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#UMAP" data-path=""><div class="tree-item-contents heading-link" heading-name="UMAP"><div class="collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div><span class="tree-item-title">UMAP</span></div></a><div class="tree-item-children nav-folder-children"><div class="tree-item" data-depth="5"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#UMAP_的基本原理" data-path=""><div class="tree-item-contents heading-link" heading-name="UMAP 的基本原理"><span class="tree-item-title">UMAP 的基本原理</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#构建高维空间的邻域图" data-path=""><div class="tree-item-contents heading-link" heading-name="构建高维空间的邻域图"><span class="tree-item-title">构建高维空间的邻域图</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#将高维邻域图嵌入到低维空间" data-path=""><div class="tree-item-contents heading-link" heading-name="将高维邻域图嵌入到低维空间"><span class="tree-item-title">将高维邻域图嵌入到低维空间</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#优化过程" data-path=""><div class="tree-item-contents heading-link" heading-name="优化过程"><span class="tree-item-title">优化过程</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#UMAP_的核心参数" data-path=""><div class="tree-item-contents heading-link" heading-name="UMAP 的核心参数"><span class="tree-item-title">UMAP 的核心参数</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#优点" data-path=""><div class="tree-item-contents heading-link" heading-name="优点"><span class="tree-item-title">优点</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#局限性" data-path=""><div class="tree-item-contents heading-link" heading-name="局限性"><span class="tree-item-title">局限性</span></div></a><div class="tree-item-children nav-folder-children"></div></div></div></div></div></div><div class="tree-item" data-depth="3"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#UMAP_与_t-SNE_的比较" data-path=""><div class="tree-item-contents heading-link" heading-name="UMAP 与 t-SNE 的比较"><span class="tree-item-title">UMAP 与 t-SNE 的比较</span></div></a><div class="tree-item-children nav-folder-children"></div></div></div></div><div class="tree-item" data-depth="2"><a class="tree-link" href="bioinfo/bioinfo-matrix-operations.html#3D_reconstruction_from_single_cell_RNA-seq" data-path=""><div class="tree-item-contents heading-link" heading-name="3D reconstruction from single cell RNA-seq"><span class="tree-item-title">3D reconstruction from single cell RNA-seq</span></div></a><div class="tree-item-children nav-folder-children"></div></div></div></div></div></div></div><script defer="">let rs = document.querySelector(".sidebar-right"); rs.classList.add("is-collapsed"); if (window.innerWidth > 768) rs.classList.remove("is-collapsed"); rs.style.setProperty("--sidebar-width", localStorage.getItem("sidebar-right-width"));</script></div></div></body></html>